---
title: "Smoking and salivary microbiota: a cross-sectional analysis of an Italian alpine population" 
subtitle: "Controls Analysis"
author: 
  - Giacomo Antonello,
  - Freida Blostein,
  - Deesha Bhaumik,
  - Elyse Davis,
  - Martin GÃ¶gele,
  - Roberto Melotti,
  - Peter Pramstaller,
  - Cristian Pattaro,
  - Nicola Segata,
  - Betsy Foxman,
  - Christian Fuchsberger
date: "`r Sys.Date()`"
output:
  pdf_document:
    toc: true
    toc_depth: 2
---

```{r, echo=FALSE}

knitr::opts_chunk$set(
  cache = FALSE,
  concordance = TRUE,
  prompt = TRUE, # fig.width=5, fig.height=5,
  out.width = "100%",
  echo = FALSE,
  warning = FALSE,
  message = FALSE,
  error = TRUE,
  tidy = FALSE,
  comment = "")

```

```{r library setup}
# from Bioconductor (install by running: BiocManager::install("x") | x = name of the package)
# BiocManager::install(c("DESeq2", "phyloseq", "microbiome"))

#github functions
# devtools::install_github("bryandmartin/corncob")
# remotes::install_github("g-antonello/gautils")
# remotes::install_github("mikemc/speedyseq")#
# devtools::install_github("gaospecial/ggvenn")

library(gautils)
library(magrittr)
library(ggpubr)
library(kableExtra)
library(microViz)

# Load control samples data
# not working yet, need to transfer from local to server

asvtable <- read_tsv("/shared/statgen/microbiome/CHRISMB/controls/controls_asv_table.tsv") %>% 
  column_to_rownames("ASV")
taxtab <- read_tsv("/shared/statgen/microbiome/CHRISMB/controls/controls_tax_table.tsv")  %>% 
  set_rownames(as.character(.$ASV))
metadata <- read_tsv("/shared/statgen/microbiome/CHRISMB/controls/controls_meta_table.tsv")

refseq <- Biostrings::readDNAStringSet(filepath = "/shared/statgen/microbiome/CHRISMB/controls/controls_refseq.phy")

# check identity of rownames and colnames. for the ASVs, the reduce/Reduce strategy did not work
samples_order_identical <- all(purrr::reduce(list(colnames(asvtable), metadata$Sample_ID), identical))
ASVs_order_identical <- all(sapply(1:nrow(expand.grid(c(1,2,3), c(1,2,3))), function(i) identical(list(rownames(asvtable), rownames(taxtab), names(refseq))[[expand.grid(c(1,2,3), c(1,2,3))[i,1]]],list(rownames(asvtable), rownames(taxtab), names(refseq))[[expand.grid(c(1,2,3), c(1,2,3))[i,2]]])) )

if(samples_order_identical & ASVs_order_identical){
  phy_only_ctrls <- phyloseq(otu_table(asvtable, taxa_are_rows = T), tax_table(as.matrix(taxtab)), sample_data(metadata %>% set_rownames(NULL)), refseq(refseq))  
} else{
  stop("Rownames or colnames of phyloseq component tables do not match")
}

phy_only_ctrls %<>% mutate_sample_data(Control_Type = case_when(ctrl_type == "DNA_extraction_ctrl" ~ "DNA extraction (Neg)",
                                                                  ctrl_type == "water_ctrl" ~ "Water (Neg)",
                                                                  ctrl_type == "zymo_pos" ~ "Zymo Mock (Pos)",
                                                                  TRUE ~ NA))

table(paste("B.", phy_only_ctrls@sam_data$batch), phy_only_ctrls@sam_data$Control_Type) %>% 
  kbl(format = "latex", caption = "Number of samples per control type (columns) in each sequencing batch (rows)") %>% 
  kable_styling(latex_options = "HOLD_position") %>% 
  row_spec(0, bold = T)
  
```

# PCoA (Beta diversity) split by controls

```{r,fig.width=8.5, fig.height=9}
# all 
theme_set(theme_light())

phy_only_ctrls_nonZero <- subset_samples(phy_only_ctrls, counts_per_sample > 0)
phy_only_ctrls_nonZero <- subset_taxa(phy_only_ctrls_nonZero, taxa_sums(phy_only_ctrls_nonZero) > 0)

distance_all <- phyloseq::distance(phy_only_ctrls_nonZero %>% phy_transform("compositional"), method = "bray")

basic_plot_all <- phy_betaPlot(phy_only_ctrls_nonZero,dist = distance_all, method = "PCoA", color = "Control_Type", palette = "jco", visual_grouping = "ellipses")+ labs(title = "All controls", color = "Control Type")
  

# only zymo
phy_only_ctrls_nonZero <- subset_samples(phy_only_ctrls, counts_per_sample > 0 & ctrl_type == "zymo_pos")
phy_only_ctrls_nonZero <- subset_taxa(phy_only_ctrls_nonZero, taxa_sums(phy_only_ctrls_nonZero) > 0)

basic_plot_zymo <- phy_betaPlot(phy_only_ctrls_nonZero,dist = distance_all, method = "PCoA", color = "batch", palette = "futurama", visual_grouping = "ellipses")+ labs(title = "Zymo", color = "Sequencing Batch")

# only extr ctrl

phy_only_ctrls_nonZero <- subset_samples(phy_only_ctrls, counts_per_sample > 0 & ctrl_type == "DNA_extraction_ctrl")
phy_only_ctrls_nonZero <- subset_taxa(phy_only_ctrls_nonZero, taxa_sums(phy_only_ctrls_nonZero) > 0)

basic_plot_DNA_extr <- phy_betaPlot(phy_only_ctrls_nonZero,dist = distance_all, method = "PCoA", color = "batch", palette = "futurama", visual_grouping = "ellipses")+ labs(title = "DNA extr. Control", color = "Sequencing Batch")

# only water ctrl

phy_only_ctrls_nonZero <- subset_samples(phy_only_ctrls, counts_per_sample > 0 & ctrl_type == "water_ctrl")
phy_only_ctrls_nonZero <- subset_taxa(phy_only_ctrls_nonZero, taxa_sums(phy_only_ctrls_nonZero) > 0)

basic_plot_water_neg <- phy_betaPlot(phy_only_ctrls_nonZero,dist = distance_all, method = "PCoA", color = "batch", palette = "futurama", visual_grouping = "ellipses")+ labs(title = "DNA extr. Control", color = "Sequencing Batch")


ggpubr::ggarrange(basic_plot_zymo, basic_plot_DNA_extr,basic_plot_water_neg, basic_plot_all,
                  
                  nrow = 2, ncol = 2, common.legend = TRUE, legend = "top", labels = "AUTO")

```

**Figure 1. Bray-Curtis distance based Principal Coordinate Analysis plots of Controls** (A) Zymo Mock community positive control. (B) Water extraction control. (C) DNA Extraction Control. The **legend on top** of the table refers to panels **A-C only** (D) Same visualization, but all control types were included, which induced a clear separation between negative and Zymo positive controls. The Bray-Curtis distance was calculated on the whole data set at first, then subset every time to keep only available samples. This was done to induce further separation based also on the taxa unique to the Zymo and the negative controls. 

# Zymo Mock communities composition (Positive Controls)

## Heatmap by ASV

```{r, fig.height=8, fig.width=8}
# Only Batches containing our study's samples

phy_ctrls_zymo_chrismb <- phy_only_ctrls %>% 
  filter_sample_data(
    ctrl_type == "zymo_pos",
    batch %in% as.character(c(1:5, 7))
    ) %>% 
  phy_pruneTaxa()

plot_chrismb_batches <- phy_ctrls_zymo_chrismb %>% 
  abundances("compositional") %>% 
  pheatmap::pheatmap(scale = "none", annotation_col = dplyr::select(meta(phy_ctrls_zymo_chrismb), batch) %>% mutate(batch = paste("B.", batch)))

```

**Figure 2. Heatmap of the relative abundance taxonomic composition at ASV level of Zymo Mock communities** No scaling except the relative abundance transformation was applied to the ASVs. Clustering was performed on both rows and columns. Columns were annotated by sequencing batch.

## Barplot by ASV

```{r, fig.width = 8, fig.height=9}
phy_ctrls_zymo_relabund <- phy_ctrls_zymo_chrismb %>% 
  mutate_sample_data(batch = paste("B.", batch)) %>% 
  tax_fix() %>% 
  comp_barplot(tax_level = "Species", 
               label = NULL,
               bar_outline_colour = "darkgray",
               facet_by = "batch") + 
  labs(
    title = "Genus",
    y = "Relative Abundance"
    ) + 
  theme(legend.position = "bottom",
        axis.ticks.x = element_blank(), panel.border = element_blank())+
  guides(fill=guide_legend(nrow=3, byrow=FALSE))
  
phy_ctrls_zymo_relabund

```

**Figure 3. Relative abundance of Species found in each Zymo Mock sample viewed as stacked barplots** Samples were separated (faceted) by sequencing batch. The legend is ordered by columns based on decreasing taxon abundance across all mock samples.

\newpage

## How much variability does the Batch explain?

```{r}
distance_zymos <- as.matrix(distance_all)[phy_ctrls_zymo_chrismb@sam_data %>% rownames(),phy_ctrls_zymo_chrismb@sam_data %>% rownames()] %>% 
  as.dist()
library(vegan)

betadisp <- betadisper(distance_zymos, group = phy_ctrls_zymo_chrismb@sam_data$batch) %>% permutest()

betadisp$tab %>% 
  kbl(format = "latex", caption = "PERMUTEST (Tukey post-hoc) with default settings on the beta-dispersity in relation to batch", digits = 3) %>% 
  kable_styling(latex_options = "HOLD_position") %>% 
  row_spec(0, bold = T)

zymo_permanova <- vegan::adonis2(formula = distance_zymos ~ plate + batch, data = meta(phy_ctrls_zymo_chrismb))

zymo_permanova %>% 
  broom::tidy() %>% 
  mutate(p.value = format.pval(p.value, digits = 1)) %>% 
  kbl(format = "latex", digits = 3, caption = "Bray-Curtis distance-based PERMANOVA with standard settings to view the variability partitioning and significance in relation to sequencing batch and DNA extraction plate for Zymo Mock communities (Positive Controls)") %>% 
  kable_styling(latex_options = "HOLD_position") %>% 
  row_spec(0, bold = T)

```
