---
title: "Smoking Paper - Main" 
subtitle: "Genus level"
author: "Giacomo Antonello"
date: "2022-08-02"
output:
  rmdformats::readthedown:
    self_contained: true
    code_folding: hide
    toc_depth: 2
    toc_float: true
    number_sections: false
    thumbnails: false
    lightbox: true
    gallery: false
    use_bookdown: true
    highlight: haddock

---

```{r, echo=FALSE}

knitr::opts_chunk$set(
  cache = FALSE,
  concordance = TRUE,
  prompt = TRUE, # fig.width=5, fig.height=5,
  out.width = "100%",
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  error = TRUE,
  tidy = FALSE,
  comment = ""
  )

```

# Data preparation {.tabset}

```{r library setup}
# from Bioconductor (install by running: BiocManager::install("x") | x = name of the package)
# BiocManager::install(c("DESeq2", "phyloseq", "microbiome"))

packages_github_links <- list(
  speedyseq = "mikemc/speedyseq",
  gautils = "g-antonello/gautils",
  LinDA = "zhouhj1994/LinDA"
)
missing_github_pkgs <- !sapply(names(packages_github_links), function(pkg) pkg %in% rownames(installed.packages()))
missing_github_pkgs <- names(missing_github_pkgs[missing_github_pkgs])

if(length(missing_github_pkgs) > 0){
sapply(missing_github_pkgs, function(pkg) devtools::install_github(packages_github_links[[pkg]]))
}

library(gautils)
library(magrittr) # imported separately from tidyverse to also import '%<>%'
library(ggpubr)
library(kableExtra)

# graphics packages
# library(ggpubr)
# library(magrittr)
# # my functions
# source("my_personal_functions.R")
# 
# 
# library(speedyseq)
theme_set(theme_light())


chrismb_phy <- readRDS("/shared/statgen/microbiome/CHRISMB/processed_data/microbiome_tables/phyloseq_built.Rds")

threshold_prevalence <- 0.01
threshold_detection <- 10




```

## Download CHRIS data

These are based on data retrieval from the `chrisData` internal R package curated by J. Rainer. Variable selection is done via consulting the codebook. 

```{r, message=FALSE, warning=FALSE}
chrismb_phy <- readRDS("/shared/statgen/microbiome/CHRISMB/processed_data/microbiome_tables/phyloseq_built.Rds")

library(chrisData)

possible_data_files <- chrisDataFile()

selected_data_files <- c(
    "general_information",
    "household",
    "clinical_traits",
    "interview",
    "labdata",
    "substudies",
    "drugs_summary"
  )

data_selected.list <- sapply(selected_data_files, function(x) chrisData(paste0(x, ".txt.gz")), USE.NAMES = T)

data_selected.joined.df <- purrr::reduce(data_selected.list, full_join, by = "AID")
data_selected.joined.df <- data_selected.joined.df[match(chrismb_phy@sam_data$AID, data_selected.joined.df$AID),]

# Columns with all NAs must be necessarily removed 
all_NAs <- apply(data_selected.joined.df, 2, function(x) all(is.na(x)))
data_selected.joined.df <- data_selected.joined.df[, !all_NAs]

# Columns of class "chron" are problematic, they must be removed to be able to work with phyloseq and metadata data.frame back and forth
data_selected.joined_NODates.df <- select(data_selected.joined.df, -contains(c("x0bp08", "x0bc03", "x0bc06")))

rm(data_selected.list)

chrismb_phy <- phy_add_metadata_variables(chrismb_phy, df = data_selected.joined_NODates.df, by = "AID")


levels(chrismb_phy@sam_data$x0oh01) <- c("0", "1-9", "10-19", "20+")
levels(chrismb_phy@sam_data$x0sm51) <- c("Never", "Former", "Current (R)", "Current (NR)")
chrismb_phy@sam_data$x0an03a %<>% as.numeric()

chrismb_phy@sam_data$x0_ager_cat <- cut(chrismb_phy@sam_data$x0_ager, breaks = c(18, 31, 41, 51, 61, 71, max(chrismb_phy@sam_data$x0_ager)+1), include.lowest = TRUE, labels = c("18-30", "31-40", "41-50", "51-60", "61-70", "71+"))

# create some househod (hid) info
chrismb_phy <- relocate_sample_data(chrismb_phy, x0_ager_cat, .after = x0_ager) %>% 
  mutate_sample_data(
    hid = paste0("h", hid),
    hid_size_chrismb = add_N_to_catVar(hid) %>% as.character() %>% gsub("h.*\\ ", "", .) %>% parse_number(),
  smoking_exposure_ga = 
    case_when(
      grepl("Current", as.character(x0sm51)) ~ "Current",
      TRUE ~ as.character(x0sm51)
      ) %>% 
    factor(levels = c("Never", "Former", "Current"))
  ) %>% 
  relocate_sample_data(smoking_exposure_ga, .before = x0sm51)

```

## Smoking variables generation

### Bin smoking intensity (g/day)

```{r}

###############################
## bin daily tobacco smoked
###############################

chrismb_phy %<>%
  mutate_sample_data(
    ### tobacco g per day (bin 5 by 5)
    x0sm61_bin5cont = case_when(
      smoking_exposure_ga == "Current" & x0sm61 <= 1 ~ 1,
      smoking_exposure_ga == "Current" & x0sm61 <= 5 ~ 5,
      smoking_exposure_ga == "Current" & x0sm61 <= 10 ~ 10,
      smoking_exposure_ga == "Current" & x0sm61 <= 15 ~ 15,
      smoking_exposure_ga == "Current" & x0sm61 <= 20 ~ 20,
      smoking_exposure_ga == "Current" & x0sm61 <= 25 ~ 25,
      smoking_exposure_ga == "Current" & x0sm61 <= 30 ~ 30,
      TRUE ~ NA
    ),
    
    x0sm61_bin2cont = case_when(
      smoking_exposure_ga == "Current" & x0sm61 <= 2 ~ 2,
      smoking_exposure_ga == "Current" & x0sm61 <= 4 ~ 4,
      smoking_exposure_ga == "Current" & x0sm61 <= 6 ~ 6,
      smoking_exposure_ga == "Current" & x0sm61 <= 8 ~ 8,
      smoking_exposure_ga == "Current" & x0sm61 <= 10 ~ 10,
      smoking_exposure_ga == "Current" & x0sm61 <= 12 ~ 12,
      smoking_exposure_ga == "Current" & x0sm61 <= 14 ~ 14,
      smoking_exposure_ga == "Current" & x0sm61 <= 16 ~ 16,
      smoking_exposure_ga == "Current" & x0sm61 <= 18 ~ 18,
      smoking_exposure_ga == "Current" & x0sm61 <= 20 ~ 20,
      smoking_exposure_ga == "Current" & x0sm61 <= 22 ~ 22,
      smoking_exposure_ga == "Current" & x0sm61 <= 24 ~ 24,
      smoking_exposure_ga == "Current" & x0sm61 <= 26 ~ 26,
      smoking_exposure_ga == "Current" & x0sm61 <= 28 ~ 28,
      smoking_exposure_ga == "Current" & x0sm61 <= 30 ~ 30,
      smoking_exposure_ga == "Current" & x0sm61 <= 60 ~ 60,
      TRUE ~ NA
    ))

```

### Bin Years since quitting

```{r}
# x0sm40 = age when people quit or reduce
# x0_ager = age rounded to closest integer

max_value <- max(chrismb_phy@sam_data$x0_ager - chrismb_phy@sam_data$x0sm40, na.rm = T)

chrismb_phy %<>% 
  mutate_sample_data(
    years_since_quitting_or_reducing = x0_ager - x0sm40,
    
    years_since_quitting_binned = case_when(
      smoking_exposure_ga == "Former" & !is.na(years_since_quitting_or_reducing) ~ cut(years_since_quitting_or_reducing, 
                                            breaks = c(0, 1, 3, 5, 10, max_value), 
      include.lowest = TRUE
    )
  )
)

rm(max_value)

```

### Ultimate smoking variable for plots: "smoking_detailed"

```{r}

max_years_since_quitting <- max(chrismb_phy@sam_data$years_since_quitting_or_reducing, na.rm = T)

chrismb_phy %<>%
  mutate_sample_data(
    smoking_detailed = case_when(
      smoking_exposure_ga == "Never" ~ "Never",
      
      smoking_exposure_ga == "Former" & years_since_quitting_or_reducing <= 1 ~ "Former 0-1 y",
      smoking_exposure_ga == "Former" & years_since_quitting_or_reducing <= 3 ~ "Former 2-3 y",
      smoking_exposure_ga == "Former" & years_since_quitting_or_reducing <= 5 ~ "Former 4-5 y",
      smoking_exposure_ga == "Former" & years_since_quitting_or_reducing <= 10 ~ "Former 6-10 y",
      smoking_exposure_ga == "Former" & years_since_quitting_or_reducing <= max_years_since_quitting ~ "Former 11-62 y",
      smoking_exposure_ga == "Former" & is.na(years_since_quitting_or_reducing) ~ "Former NA y",
      
      smoking_exposure_ga == "Current" & x0sm61 < 2 ~ "Current < 2 g",
      smoking_exposure_ga == "Current" & x0sm61 <= 3 ~ "Current 2-3 g",
      smoking_exposure_ga == "Current" & x0sm61 <= 5 ~ "Current 4-5 g",
      smoking_exposure_ga == "Current" & x0sm61 <= 10 ~ "Current 6-10 g",
      smoking_exposure_ga == "Current" & x0sm61 <= 15 ~ "Current 11-15 g",
      smoking_exposure_ga == "Current" & x0sm61 <= 20 ~ "Current 16-20 g",
      smoking_exposure_ga == "Current" & x0sm61 > 20 ~ "Current > 20 g",
      smoking_exposure_ga == "Current" & is.na(x0sm61) ~ "Current NA g",
      TRUE ~ "Unknown"
      ),
    
    smoking_detailed = factor(smoking_detailed, levels = unique(smoking_detailed)[c(14, 4, 10, 6, 3, 7, 12, 8, 11, 13, 5, 9, 2, 16, 1, 15)])
    )

```

## Available Samples for statistical analysis

```{r}

excluded_samples_upon_request <- readLines("/home/gantonello/CHRISMB/participants lists each study/withdrawnConsents_AIDs.txt")

chrismb_phy_core <- core(chrismb_phy, detection = threshold_detection, prevalence = threshold_prevalence)

phy_Q1 <- chrismb_phy_core %>% 
  filter_sample_data(!(AID %in% excluded_samples_upon_request),
                     !is.na(x0oh01),
                     !is.na(x0_ager),
                     !is.na(x0_sex),
                     !is.na(x0sm51),
                     #x0bc10 != 1, # people who smoked were reported as 1, NAs are treated as non-smoking
                     x0dd51 == "No")

phy_Q1


phy_Q1_genus <- phy_tax_glom(phy_Q1, "Genus")


phy_Q1_meta <- meta(phy_Q1)

```

```{r}
dir.create("../results")
```

# Table 1

## Table shorter

```{r}

library(table1)

tbl1 <- phy_Q1_meta %>% 
  # do some renaming, just for the Table output
  dplyr::rename(
    Sex = x0_sex,
    `Age Category (years)` = x0_ager_cat, 
    `N° Teeth (self-reported)` = x0oh01, 
    `Gums Health (self-reported)` = x0oh02b
  ) %>% 
# pipe this temporary data into the table 1
  table1(~ Sex + `Age Category (years)` + `N° Teeth (self-reported)` + `Gums Health (self-reported)`| x0sm51, 
       data = ., 
       extra.col = list("X² p-value" = table1.pvalues),
       caption = "Demographics of CHRISMB cohort, in South Tyrol, Italy, with respect to smoking habit. Per-column percentages were also reported in brackets. Current smokers were separated into smokers who reduced daily smoking dosage some time in the past - Current (R), and those who did not reduce - Current (NR). The whole cohort is included under the “CHRISMB” column. Significance is calculated as X² test for categorical variables and as Kruskal-Wallis non-parametric analysis of variance for continuous variables (Blood pressure, Body Mass Index",
       #overall = "CHRISMB"
       )

tbl1
```

## Table more variables

```{r}
tbl2 <- phy_Q1_meta %>% 
  # do some renaming, just for the Table output
  dplyr::rename(
    Sex = x0_sex,
    `Age Category (years)` = x0_ager_cat, 
    `N° Teeth (self-reported)` = x0oh01, 
    `Gums Health (self-reported)` = x0oh02b,
    `Systolic Blood Pressure (mmHg)` = x0bp01,
    `Diastolic Blood Pressure (mmHg)` = x0bp02,
    `Body Mass Index (kg/m²)` = x0an03a
  ) %>% 
# pipe this temporary data into the table 1
  table1(~ Sex + `Age Category (years)` + `N° Teeth (self-reported)` + `Gums Health (self-reported)` + `Systolic Blood Pressure (mmHg)` + `Diastolic Blood Pressure (mmHg)` + `Body Mass Index (kg/m²)`| x0sm51, 
       data = ., 
       extra.col = list("P-value" = table1.pvalues),
       caption = "Demographics of CHRISMB cohort, in South Tyrol, Italy, with respect to smoking habit. Per-column percentages were also reported in brackets. Current smokers were separated into smokers who reduced daily smoking dosage some time in the past - Current (R), and those who did not reduce - Current (NR). The whole cohort is included under the “CHRISMB” column. Abbreviations: mmHg = millimeters of mercury, kg = kilograms, m² square meters. Significance is calculated as X² test for categorical variables and as Kruskal-Wallis non-parametric analysis of variance for continuous variables (Blood pressure, Body Mass Index",
       #overall = "CHRISMB"
       )

tbl2

```

# Figure 1 - Microbiota in relation to smoking status {.tabset}

## Figure 1A

```{r}

div <- phyloseq::distance(microbiome::transform(phy_Q1_genus, "compositional"), "bray")

ord <- ordinate(physeq = phy_Q1_genus,
                distance = div,
                method = "PCoA"
)

fig1A <- plot_ordination(physeq = phy_Q1_genus,
                         ordination = ord, 
                         color = "smoking_exposure_ga")+ 
  labs(title = NULL)+
  theme_light()+
  theme(legend.position = "bottom")+ 
  scale_color_discrete(name = "Smoking habit",type = ggsci::pal_jco("default")(4)[c(1,2,4)])+ 
  stat_ellipse(aes(group = smoking_exposure_ga), show.legend = FALSE)+
  guides(color = guide_legend(override.aes = list(size=3)))

fig1A$labels$caption <- gsub(fig1A$labels$caption, pattern = "\n", replacement = "; ")
fig1A$labels$x <- paste0("Principal Coordinate 1   ", strsplit(fig1A$labels$x, "   ")[[1]][[2]])
fig1A$labels$y <- paste0("Principal Coordinate 2   ", strsplit(fig1A$labels$y, "   ")[[1]][[2]])

```

## Figure 1B {.tabset}

```{r}
dir.create("../results/Figure1/DiffAbundGenera/", recursive = T)
```

### Diff abund 1: DESeq2

```{r, fig.align='center'}

dir.create("../results/Figure1/DiffAbundGenera/DESeq2/", recursive = T)

if(!file.exists("../results/Figure1/DiffAbundGenera/DESeq2/figure1_deseq2_dataframe.list.RDS")){
library(DESeq2)
#############################################################################

deseq_object_genus <-
  phyloseq_to_deseq2(phy_Q1_genus, design = ~ x0_sex + x0_ager_cat + x0oh01 + smoking_exposure_ga)

geom_means <- apply(counts(deseq_object_genus), 1, function(x) exp(mean(log(x[x>0]))))

# estimate size factors, important to account for library size variations
estim_size <- estimateSizeFactors(deseq_object_genus, geoMeans = geom_means)

# compute the DESeq2 glm, accounting for the estimate size factor and geometric means

deseq2_results_genus <- DESeq(estim_size, fitType = "local")


saveRDS(deseq2_results_genus, "../results/Figure1/DiffAbundGenera/DESeq2/figure1_deseq2_dataframe.list.RDS")
} else{
  deseq2_results_genus <- readRDS("../results/Figure1/DiffAbundGenera/DESeq2/figure1_deseq2_dataframe.list.RDS")
}

```

### Diff abund 2: LinDA

```{r, include=FALSE}
#devtools::install_github("zhouhj1994/LinDA")

dir.create("../results/Figure1/DiffAbundGenera/LinDA/", recursive = T)

if(!file.exists("../results/Figure1/DiffAbundGenera/LinDA/LinDA_output.list.RDS")){
  
library(LinDA)

LinDA_model_genera <- linda(otu.tab = abundances(phy_Q1_genus),
                   meta = meta(phy_Q1_genus), 
                   formula =  "~ x0_sex + x0_ager_cat + x0oh01 + smoking_exposure_ga", 
                   alpha = 0.05,
                   prev.cut = 0.01, 
                   lib.cut = 1000, 
                   winsor.quan = 0.97,
                   adaptive = T)

saveRDS(LinDA_model_genera, "../results/Figure1/DiffAbundGenera/LinDA/LinDA_output.list.RDS")
}else{
  LinDA_model_genera <- readRDS("../results/Figure1/DiffAbundGenera/LinDA/LinDA_output.list.RDS")
}

```


### Diff abund 3: Maaslin2

```{r, include=FALSE}

dir.create("../results/Figure1/DiffAbundGenera/Maaslin2", showWarnings = FALSE, recursive = T)


if(!file.exists("../results/Figure1/DiffAbundGenera/Maaslin2/significant_results.tsv")){
  
  library(Maaslin2)
 # older normaliz and so on 
  # Maaslin2(
  #   input_data = picrust_taxa_matrix_Q1, 
  #   input_metadata = phy_Q1_meta, 
  #   output = "../results/Maaslin2_DA", 
  #   analysis_method = "LM",
  #   fixed_effects = c("age_cat", "sex", "x0oh01", "smoking_exposure_ga"),
  #   reference = c("smoking_exposure_ga, Never"),
  #   transform = "LOG",
  #   normalization = "TMM",
  #   correction = "BH",
  #   plot_heatmap = F
  # )

  # these following are the same settings used by Nearing  et al. (2021) 
  Maaslin2(
    input_data = abundances(phy_Q1_genus), 
    input_metadata = meta(phy_Q1_genus), 
    output = "../results/Figure1/DiffAbundGenera/Maaslin2/", 
    analysis_method = "LM",
    fixed_effects = c("x0_sex", "x0_ager_cat", "x0oh01","smoking_exposure_ga"),
    reference = "smoking_exposure_ga, Never",
    transform = "LOG",
    normalization = "TSS",
    correction = "BH",
    standardize = FALSE,
    plot_heatmap = F
  )
  

}

genera_names_map_all <- as.data.frame(tax_table(phy_Q1_genus))["Genus"] %>% 
  mutate(feature = make.names(Genus))

maaslin_results_genera <- read_tsv("../results/Figure1/DiffAbundGenera/Maaslin2/significant_results.tsv") %>% 
  left_join(genera_names_map_all, by = "feature") %>% 
  mutate(feature = Genus) %>% 
  dplyr::select(-Genus)

```

### Diff abund 4: ALDEx2 (glm)

```{r}
dir.create("../results/Figure1/DiffAbundGenera/ALDEx2/", recursive = T)

if (!file.exists("../results/Figure1/DiffAbundGenera/ALDEx2/ALDEx2.DA.rawResults_genera.list.RDS")) {
  
  library(ALDEx2)
  # define model matrix, based on the metadata
  model_mtx <-
    model.matrix(
      ~ x0_sex + x0_ager_cat + x0oh01 + smoking_exposure_ga,
      data = dplyr::select(
        meta(phy_Q1_genus),
        x0oh01,
        x0_sex,
        x0_ager_cat,
        smoking_exposure_ga
      )
    )
  # perform the clr differential abundance
  aldex.glm.genera_smoking_ga <-
    aldex.clr(abundances(phy_Q1_genus),
              model_mtx,
              mc.samples = 200,
              denom = "all", 
              useMC = T
              )
  
  # make glm, obtain significance of terms
  glm.test_genera_smoking_ga <- aldex.glm(aldex.glm.genera_smoking_ga, model_mtx)
  
  # this step gets estimates effect sizes, it takes ages, like 30-60 minutes
  glm.effects_genera_smoking_ga <- aldex.glm.effect(aldex.glm.genera_smoking_ga, include.sample.summary = T, verbose = FALSE, useMC = T)
  
  # aggregate these results into one data frame, that looks like lme4 output
  aldex.results.genera <- list(clr.model = aldex.glm.genera_smoking_ga,
                        glm.test = glm.test_genera_smoking_ga,
                        eff_size.test = glm.effects_genera_smoking_ga)
  
  
  saveRDS(aldex.results.genera, "../results/Figure1/DiffAbundGenera/ALDEx2/ALDEx2.DA.rawResults_genera.list.RDS")
  
} else {
  aldex.results.genera <- readRDS("../results/Figure1/DiffAbundGenera/ALDEx2/ALDEx2.DA.rawResults_genera.list.RDS")
}

```

### Diff abund 5: ANCOM-BC

```{r}
dir.create("../results/Figure1/DiffAbundGenera/ANCOM-BC/", recursive = T)

if (!file.exists("../results/Figure1/DiffAbundGenera/ANCOM-BC/ANCOM-BC.DA.rawResults_genera.list.RDS")) {
tmp <- mia::makeTreeSummarizedExperimentFromPhyloseq(phy_Q1_genus %>% select_sample_data(x0_sex, x0_ager_cat, x0oh01, smoking_exposure_ga)) 


library(ANCOMBC)

# takes around 30 minutes
ancom_results <- ancombc2(data = tmp,
        assay_name = "counts", 
        tax_level = NULL,
        fix_formula = "x0_sex + x0_ager_cat + x0oh01 + smoking_exposure_ga", 
        group = "smoking_exposure_ga",
        struc_zero = TRUE,
        iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
        em_control = list(tol = 1e-5, max_iter = 20),  
        p_adj_method = "BH",
        n_cl = 1,
        alpha = 0.05)

saveRDS(ancom_results, "../results/Figure1/DiffAbundGenera/ANCOM-BC/ANCOM-BC.DA.rawResults_genera.list.RDS")
detach("package:ANCOMBC", unload = T)
}else{
  
  ancom_results <- readRDS("../results/Figure1/DiffAbundGenera/ANCOM-BC/ANCOM-BC.DA.rawResults_genera.list.RDS")
  
}

```

### Summary of results

Results filtered by two conditions:

  1. FDR (5%) corrected $q-value < 0.05$
  2. The effect size, irrespectively of the way it is estimated, must be larger than 0.5

Since this value is rather arbitrary, one could devise a *quantile cutoff*, but it's hard to decide where to cut, since we have no *a priori* expectations of the effect size distributions.
  
```{r, include=FALSE}

library(DESeq2)

# DESeq2
deseq2_signif_results <- results(deseq2_results_genus, 
                                 contrast = c("smoking_exposure_ga","Current", "Never"),
                                 pAdjustMethod = "BH") %>% 
  as.data.frame() %>% 
  rownames_to_column("taxon") %>% 
  filter(padj < 0.05)

# LinDA
LinDAresults_taxon_signif_Current <- LinDA_model_genera$output$smoking_exposure_gaCurrent %>% 
  filter(padj < 0.05, 
         #abs(log2FoldChange) > 0.5
         ) %>% 
  rownames_to_column("taxon")

# MaAsLin2
maaslin_signif <- maaslin_results_genera %>% 
  filter(metadata == "smoking_exposure_ga") %>% 
  dplyr::select(feature, value, coef, stderr, pval, qval, N.not.0) %>% 
  filter(qval < 0.05, 
         #abs(coef) > 0.5, 
         value == "Current")

# ALDEx2
aldex_gLinDAsmoking_signif <- aldex.results.genera$glm.test %>% 
  as.data.frame() %>% 
  dplyr::select(contains("smoking_exposure_gaCurrent")) %>% 
  dplyr::rename_all(.funs = function(x) gsub(pattern =  ".*\\:", replacement = "", x = x)) %>% 
  rownames_to_column("taxon") %>% 
  as_tibble() %>% 
  filter(
    #abs(Est) > 0.5, 
    pval.holm < 0.05)
  
# ANCOM-BC

ancom2_signif_results <- ancom_results$res %>% 
  dplyr::select(contains(c("taxon", "gaCurrent")), -contains("diff_")) %>% 
  rownames_to_column("taxon") %>% 
  set_names(c("taxon", "log2FoldChange", "lfcSE", "wald_stat", "p.value", "padj")) %>% 
  filter(
    #abs(log2FoldChange) > 0.5,
    padj < 0.05
    )

all_models_signif_genera <- list(ALDEx2 = aldex_gLinDAsmoking_signif %>% dplyr::select(taxon, Est) %>% set_names(c("taxon", "ALDEx2")), 
                    MaAsLin2 = maaslin_signif %>% dplyr::select(feature, coef) %>% set_names(c("taxon", "MaAsLin2")), 
                    LinDA = LinDAresults_taxon_signif_Current %>% dplyr::select(taxon,log2FoldChange) %>% set_names(c("taxon", "LinDA")), 
                    DESeq2 = deseq2_signif_results %>% dplyr::select(taxon, log2FoldChange) %>% set_names(c("taxon", "DESeq2")),
                    `ANCOM-BC` = ancom2_signif_results %>% dplyr::select(taxon, log2FoldChange) %>% set_names(c("taxon", "ANCOM-BC"))
                    ) %>% 
  purrr::reduce(., full_join, by = "taxon")


frequency_signif_taxon <- pivot_longer(all_models_signif_genera, cols = !starts_with("taxon"), names_to = "DA_method", values_to = "da_eff_size") %>% 
  filter(!is.na(da_eff_size)) %>% 
  group_by(taxon) %>% 
  tally() %>% 
  arrange(desc(n))


taxa_union <- all_models_signif_genera$taxon

taxa_consensus <- frequency_signif_taxon %>% 
  filter(n >= 4) %>% 
  .$taxon

taxa_intersect <- all_models_signif_genera %>% 
  .[complete.cases(.),] %>%
  .$taxon

saveRDS(taxa_union, "../results/Figure1/DiffAbundGenera/significant_genera_union.RDS")
saveRDS(taxa_intersect, "../results/Figure1/DiffAbundGenera/significant_genera_intersect.RDS")
saveRDS(taxa_consensus,  "../results/Figure1/DiffAbundGenera/significant_genera_consensus.RDS")

saveRDS(all_models_signif_genera, "../results/Figure1/DiffAbundGenera/significant_genera_Consensus-and-log2FC.RDS")

ggvenn::ggvenn(list(
  #"Union" = taxa_union,
  "Consensus" = taxa_consensus,
  "Intersection" = taxa_intersect)
)

taxa_union <- readRDS("../results/Figure1/DiffAbundGenera/significant_genera_union.RDS")

```

```{r, include=FALSE}

fig1B <- phy_Q1_genus %>%
  mutate_sample_data(smoking_exposure_ga = as.character(add_N_to_catVar(smoking_exposure_ga))) %>% 
  microbiome::transform("compositional") %>% 
  filter_tax_table(taxa_names(.) %in% taxa_consensus) %>% 
  phy_summarise_taxa_by_metadata("smoking_exposure_ga") %>% 
  pheatmap::pheatmap(angle_col = 315, scale = "row", treeheight_col = 0, treeheight_row = 0)

fig1B <- ggplotify::as.ggplot(fig1B)

```

### Consensus of methods, quick discussion

```{r, include=FALSE}

NA_value <- -5

heatmap_plot <- ggplotify::as.ggplot(all_models_signif_genera %>% 
  column_to_rownames("taxon") %>% 
  mutate_all(.funs = function(x) ifelse(is.na(x), NA_value, x)) %>% 
  as.matrix() %>% 
  pheatmap::pheatmap(angle_col = 315, treeheight_row = 0, treeheight_col = 9, scale = "none")
)

plot_correlations_eff.sizes <- grid::grid.grabExpr(
  print(all_models_signif_genera %>% 
          filter(complete.cases(.)) %>% 
  column_to_rownames("taxon") %>% 
 mutate_all(.funs = function(x) ifelse(is.na(x), 0, x)) %>%
 #as.matrix() %>%
  GGally::ggpairs(title = "Correlations between effect sizes\nof the features identified in all methods")))

library(ggVennDiagram)

datavenn <- list(ALDEx2 = aldex_gLinDAsmoking_signif %>% dplyr::select(taxon) %>% set_names(c("taxon", "ALDEx2")), 
                               MaAsLin2 = maaslin_signif %>% dplyr::select(feature, coef) %>% set_names(c("taxon", "MaAsLin2")), 
                               LinDA = LinDAresults_taxon_signif_Current %>%  dplyr::select(taxon,log2FoldChange) %>% set_names(c("taxon", "LinDA")), 
                               DESeq2 = deseq2_signif_results %>% dplyr::select(taxon, log2FoldChange) %>% set_names(c("taxon", "DESeq2"))
                               ) %>% lapply("[[", "taxon") %>% 
  Venn() %>% 
  process_data()


venn_plot <- ggplot()+
  geom_sf(aes(fill = count), data = venn_region(datavenn)) + 
  scale_fill_gradient2(
    low = "white",
    high = scales::muted("red")
  )+
  geom_sf(color="darkgrey", size = 2, data = ggVennDiagram::venn_setedge(datavenn))+
  geom_sf_text(aes(label = name), color = "black", fontface = "bold", data = venn_setlabel(datavenn)) +  
  # add a alternative region name
  geom_sf_label(aes(label = paste0(count, "\n", round(count/sum(count)*100, 1), "%")), 
                data = venn_region(datavenn), 
                alpha = 0.5, label.size = NA) +  
  theme_void()+
  theme(legend.position = "none")

```

```{r, fig.height=10, fig.width=12}

ggarrange(ggarrange(plot_correlations_eff.sizes, 
                    venn_plot+
                      scale_x_continuous(expand = expansion(mult = .2)), 
                    
                    nrow = 2, ncol = 1, heights = c(1,0.5)
                    ), 
          
          heatmap_plot, 
          
          nrow = 1, ncol = 2
          ) %>% 
  annotate_figure(top = text_grob("Discussion of differential abundance\nresults similarity across methods", size = 16, face = "bold", color = "red2"),
                  bottom = text_grob(paste("NAs in the heatmap were arbitrarily set to", NA_value, "in order not to fail clustering,\nwhile still keeping evident track of where a method did not return a pathway as significant"), size = 10))

```

## Figure 1C - Oxygen Metabolism

```{r}
dir.create("../results/Figure1/OxygenMetabolism/", recursive = T)

otutab_with_genus_name <- phy_Q1_genus %>%
  microbiome::transform("compositional") %>%
  abundances() %>% 
  as.data.frame() %>% 
  rownames_to_column("Genus") %>% # next step helps assigning the highest number of genera/families possible (because among the taxa in the table there are some families too) 
  mutate(Genus_easier = sapply(strsplit(as.character(Genus), "_"), function(g) g[[1]]))

# get the metadata of the oxygen metabolism vs Genus
oxygen_requirement_taxa <- read.csv("https://raw.githubusercontent.com/mcalgaro93/sc2meta/master/data/genera_methabolism.tsv", sep = "\t") %>% 
  set_names(c("Genus_easier", "oxygen_requirements")) %>% 
  mutate(oxygen_requirements = gsub("F ", "Facultative ", oxygen_requirements, fixed = T))

# merge otu table and oxygen requirement data by simplified genus name
molten_df_merged_oxygen <- merge(otutab_with_genus_name, oxygen_requirement_taxa, by = "Genus_easier", all.x = TRUE)
molten_df_merged_oxygen$oxygen_requirements <- ifelse(is.na(molten_df_merged_oxygen$oxygen_requirements), "NA",molten_df_merged_oxygen$oxygen_requirements)

# aggregate by oxygen metabolism, instead of genera, to make code easier

tmp <- molten_df_merged_oxygen %>% 
  dplyr::select(!contains("Genus")) %>%
  group_by(oxygen_requirements) %>% 
  summarise_all(sum) %>%
  column_to_rownames("oxygen_requirements") %>% 
  t() %>% as.data.frame() %>% 
  rownames_to_column("sample_name")

oxygen_molten_metadata <- tmp %>% reshape2::melt() %>% 
  merge(., phy_Q1_meta, all.x = TRUE, by = "sample_name") %>% 
  dplyr::rename(oxygen = variable)

saveRDS(oxygen_molten_metadata, "../results/Figure1/OxygenMetabolism/oxygen_with_metadata_df.RDS")

```

```{r}
library(rstatix)

signif <- filter(oxygen_molten_metadata, oxygen != "NA") %>%
  mutate(aerobiosis = oxygen) %>% 
  group_by(oxygen) %>% 
  rstatix::wilcox_test(value ~ smoking_exposure_ga) %>% 
  adjust_pvalue(method = "BH") %>% 
  #add_significance("p.adj") %>% 
  add_xy_position(x = "oxygen",step.increase = 0.05) %>% 
  filter(p.adj < 0.05)

fig1C <- ggboxplot(data = filter(oxygen_molten_metadata, oxygen != "NA"),
                   x = "oxygen", 
                   y = "value", 
                   fill = "smoking_exposure_ga") + 
  ylab("Relative Abundance") + 
  xlab("")+ 
  scale_fill_discrete(name = "Smoking habit", type = ggsci::pal_jco()(4)[c(1,2,4)]) + 
  stat_pvalue_manual(signif, tip.length = 0.01) + 
  theme_light()+
  theme(legend.position = "none")
```

# {-}

## Figure 1 Panel

```{r, fig.width= 11,fig.height=10}
Fig1_ggpanel <- ggarrange(
  ggarrange(fig1A, 
            fig1C, 
            
            nrow = 2, ncol = 1, labels = c("A","C"), heights = c(1,0.8)
            ), 
  ggplot() + theme_void(),
  fig1B, 
  
  nrow = 1, ncol = 3, labels = c("", "B", ""), widths = c(1,0.1,1)
  )

ggsave(
  plot = Fig1_ggpanel,
  filename = "../results/Figure1/Figure 1.png",
  height = 7,
  width = 8,
  units = "in",
  dpi = 600
)

Fig1_ggpanel

```

# Figure 2 - Current smokers, regression vs dosage per day {.tabset}

```{r, message = FALSE}

phy_Q2 <- phy_Q1 %>% 
  filter_sample_data(smoking_exposure_ga == "Current") %>%
  filter_sample_data(!is.na(x0sm61)) %>%
  filter_sample_data(x0sm61 < 60)

phy_Q2 <- subset_taxa(phy_Q2, taxa_sums(phy_Q2)>0)

phy_Q2_genus <- phy_tax_glom(phy_Q2, "Genus")

dir.create("../results/Figure2", recursive = TRUE, showWarnings = FALSE)

phy_Q2_genus

```

## Regression

```{r, message = FALSE, warning=FALSE}

if(!file.exists("../results/Figure2/deseq_smoking_regression_datafames.rds")){
deseq_raw_results_Q2 <- list()
for(gr in c("x0sm61_bin5cont", "x0sm61_bin2cont", "x0sm61")){
  
  # create deseq object
  fla <- as.formula(paste0("~ x0_ager + x0_sex + x0oh01 + ", gr))
  
  phy_q2_genus_deseq_q2 <- phyloseq_to_deseq2(phy_Q2_genus, design = fla)
  
  # calculate geometric means, important to center the counts
  
  geom_means_q2 <- apply(counts(phy_q2_genus_deseq_q2), 1, function(x)
      exp(mean(log(x[x > 0]))))
  
  # estimate size factors, important to account for library size variations
  estim_size_q2 <- estimateSizeFactors(phy_q2_genus_deseq_q2, geoMeans = geom_means_q2)
  
  # compute the DESeq2 glm, accounting for the estimate size factor and geometric means
  
  deseq2_results_q2 <- DESeq(estim_size_q2, fitType="local")
  deseq_raw_results_Q2[[gr]] <- deseq2_results_q2
}

taxonomy <- as.data.frame(tax_table(phy_Q2_genus))[c("Genus", "Phylum")]
q2_results <- lapply(deseq_raw_results_Q2, function(res) results(res) %>% 
                       as.data.frame() %>% 
                       rownames_to_column("Genus") %>%
                       merge(taxonomy, by = "Genus", sort = FALSE, all.x = TRUE)
)

saveRDS(q2_results, "../results/Figure2/deseq_smoking_regression_datafames.rds")
rm(taxonomy)

} else{
  
  q2_results <- readRDS("../results/Figure2/deseq_smoking_regression_datafames.rds")
}


```

## Figure 2A - Heatmap

```{r, include=FALSE}

q2_results_filtered <- q2_results %>% 
  .$x0sm61_bin5cont %>% 
  as.data.frame() %>% 
  filter(padj < 0.05)

Fig2A <- phy_Q2_genus %>%
  microbiome::transform("compositional") %>% 
  #filter_sample_data(x0oh01 == "20+") %>% 
  mutate_sample_data(smoking_detailed = as.character(add_N_to_catVar(smoking_detailed))) %>% 
  filter_tax_table(taxa_names(.) %in% q2_results_filtered$Genus) %>%
  phy_summarise_taxa_by_metadata("smoking_detailed", .fun = mean) %>% 
  pheatmap::pheatmap(angle_col = 315, scale = "row", cluster_cols = TRUE,  treeheight_col = 15, treeheight_row = 0)

```

## Figure 2 B,C,D

```{r}

Fig2BCD <-
  filter(
    oxygen_molten_metadata,
    oxygen != "NA" &
      smoking_exposure_ga == "Current",
    x0sm61 < 60
  ) %>%
  group_split(oxygen) %>%
  lapply(
    function(ox)
      ggscatter(ox,
                x = "x0sm61",
                y = "value",
                alpha = 0.5,
                color = "gray30") +
      geom_smooth(se = TRUE) +
      theme_light()+
      labs(
        x = "Tobacco smoked per day (g)",
        y = "Relative Abundance",
        title = unique(ox$oxygen) 
        )
)
```


# {-}

## Figure 2 - Panel

```{r, message = FALSE, warning=FALSE, fig.height=8, fig.width=10}

# Fig2_ggpanel <- ggarrange(
#   ggplot()+ theme_void(),
#   ggplotify::as.ggplot(Fig2A), 
#   ggarrange(
#     Fig2BCD[[1]], 
#     Fig2BCD[[2]], 
#     Fig2BCD[[3]], 
#     
#     nrow = 3, ncol = 1, labels = c("B", "C", "D")
#     ), 
#   
#   nrow = 1, ncol = 3, widths = c(0.1, 1.5,1), labels = c("A",  "", "")
#   )

Fig2_ggpanel <- ggarrange(
  #ggplot()+ theme_void(),
  ggplotify::as.ggplot(Fig2A), 
  ggarrange(
    Fig2BCD[[1]], 
    Fig2BCD[[2]], 
    Fig2BCD[[3]], 
    
    nrow = 3, ncol = 1, labels = c("B", "C", "D")
    ), 
  
  nrow = 1, ncol = 2, 
  widths = c(1.5,1), 
  labels = c("A", "")
  )


ggsave(
  plot = Fig2_ggpanel,
  filename ="../results/Figure2/Figure 2.png",
  height = 7,
  width = 8,
  units = "in",
  dpi = 600
)

Fig2_ggpanel

```

# Figure 3 - Former smokers, regression vs years since quitting {.tabset}

```{r}
dir.create("../results/Figure3/")
```

```{r, message=FALSE}

phy_Q3 <- filter_sample_data(phy_Q1, 
                             smoking_exposure_ga == "Former", 
                             !is.na(years_since_quitting_or_reducing), 
                             x0sm40 >= x0sm34 # meaning: (age when they quit) >= (age when they started)
                             )

phy_Q3 <- subset_taxa(phy_Q3, taxa_sums(phy_Q3) > 0)

phy_Q3@sam_data$years_since_quitting_or_reducing %<>% as.numeric()

phy_Q3_genus <- phy_tax_glom(phy_Q3, "Genus")

phy_Q3

```

## Regression calculations and summary table

```{r, message=FALSE, warning=FALSE}

library(DESeq2)

phy_q3_genus_deseq <- phyloseq_to_deseq2(phy_Q3_genus, design = ~ x0_ager_cat + x0_sex + x0oh01 + years_since_quitting_or_reducing)

# calculate geometric means, important to center the counts
geom_means_q3 <- apply(counts(phy_q3_genus_deseq), 1, function(x)
      exp(mean(log(x[x > 0]))))

# estimate size factors, important to account for library size variations
estim_size_q3 <- estimateSizeFactors(phy_q3_genus_deseq, geoMeans = geom_means_q3)

# compute the DESeq2 glm, accounting for the estimate size factor and geometric means

deseq2_results_q3 <- DESeq(estim_size_q3, fitType="local")

results_q3 <- results(deseq2_results_q3, name = "years_since_quitting_or_reducing")

results_q3.df <- as.data.frame(results_q3)

DT::datatable(arrange(results_q3.df, padj) %>% mutate_if(is.numeric, round, 3), caption = "Table 'Years since quitting' | Results of regression of microbial genera in response to the years since quitting smoking. The model was corrected for age, sex and number of teeth, nominal P-values correceted for False Discovery Rate with the Benjamini-Hochberg method.")
```

## Figure 3A - Heatmap

```{r, include=FALSE, echo=FALSE}
caption <- "Figure 3. Smoking cessation induces a perturbation of the salivary microbiota composition, which tends to resemble never smokers’ profiles within 5 years.
(A) Heatmap of the mean relative abundance of genera significantly affected by smoking (see Figure 1), in relation to the years since quitting. Taxa were transformed to relative abundance and scaled by row, to highlight differences in mean abundance in relation to the years since quitting grouped to limit the low sample size of some categories. (B, C, D) Relative abundance of anaerobes, aerobes and facultative anaerobes in relation to years since quitting smoking"
```

```{r, fig.cap=print(caption), include=FALSE}

fig3A <- phy_Q1_genus %>% 
  microbiome::transform("compositional") %>% 
  filter_sample_data(!(grepl("NA", smoking_detailed,fixed = T)) & 
                       x0oh01 == "20+" &
                       smoking_exposure_ga != "Current"
                       ) %>% 
  mutate_sample_data(smoking_detailed = as.character(add_N_to_catVar(smoking_detailed))) %>%
  filter_tax_table(taxa_names(.) %in% taxa_consensus) %>% 
  phy_summarise_taxa_by_metadata("smoking_detailed", .fun = mean) %>% 
  pheatmap::pheatmap(angle_col = 315, scale = "row", treeheight_col = 15, treeheight_row = 0)

```

## Figure 3 B,C,D

```{r}

fig3BCD <-
  filter(
    oxygen_molten_metadata,
    oxygen != "NA" &
      sample_name %in% sample_names(phy_Q3_genus) &
    years_since_quitting_or_reducing <= 40
  ) %>%
  group_split(oxygen) %>%
  lapply(
    function(ox)
      ggscatter(ox,
                x = "years_since_quitting_or_reducing",
                y = "value",
                alpha = 0.5
      ) +
      geom_smooth(se = TRUE) +
      xlab("Years since quitting smoking") +
      ylab("Relative Abundance") +
      ggtitle(unique(ox$oxygen)) +
      
      theme_light()+geom_point(alpha = 0.1, color = "gray50")
  )

```


# {-}

## Figure 3 - Panel

```{r, fig.height=10, fig.width=10}

Fig3_ggpanel <- ggarrange(
  ggplotify::as.ggplot(fig3A), 
  ggarrange(fig3BCD[[1]], fig3BCD[[2]], fig3BCD[[3]], 
            
            nrow = 3, ncol = 1, labels = c("B", "C", "D")
  ), 
  
  nrow = 1, ncol = 2, widths = c(1.5,1), labels = c("A", "")
  )


ggsave(
  plot = Fig3_ggpanel,
  filename = "../results/Figure3/Figure 3.png",
  height = 7,
  width = 8,
  units = "in",
  dpi = 600
)

Fig3_ggpanel

```

# Figure 4 - Pathways differential abundance {.tabset}

PICRUSt2 input data were generated using the full CHRISMB dataset filtered for prevalence and detection (resp. `1%` and `10` counts), which had 623 features for 1928 samples. I though I would do this for all samples once and for all, the set of ASVs doesn't change if I pick the smoking paper subset or the CHRISMB full one. The discriminative part is the set of ASVs.

Results are stored in `/home/gantonello/CHRISMB/PICRUSt2_prediction_v2.5/` subdivided into stratified and unstratified. the R script to generate the input data is in each of those two base directories.

Then I imported the two informative datasets:

 - EC codes per sample (like looking at single enzyme level)
 - pathway abundances (like looking at the aggregated enzymes)


**HOW DOES PICRUSt2 ASSIGN EC AND PATHWAYS?** It has some mapping files inside the environment.

If you run the `picrust_pipeline.py` with default settings, PICRUSt2 will take information from:

`/home/gantonello/miniconda3/envs/picrust2/lib/python3.8/site-packages/picrust2/`
And from there it will:

   1. Assign the ASVs reference seqs to the closest bacterial genome
   2. If the distance from the closest annotated reference genome is less than 2, it assumes identity and will assign a set of potential functions using `./default_files/prokaryotic/ec.txt.gz`
   3. EC numbers are converted into enzyme names with `./default_files/pathway_mapfiles/ec_level4_to_metacyc_rxn.tsv`
   4. these enzyme names are assigned to pathway using `./default_files/pathway_mapfiles/metacyc_path2rxn_struc_filt_pro.txt`

Then the names get nicer names with a simple, final function, and that's what I imported for the analysis.

Let's start with pathways, correcting for **age category, sex, and N° teeth**, as the rest of the paper draft

```{r}
dir.create("../results/Figure4/", showWarnings = F)
```

```{r pathways_data_loading}

picrust_pathways_matrix <- data.table::fread("/home/gantonello/CHRISMB/PICRUSt2_prediction_v2.5/unstratified/picrust2_output/pathways_out/path_abun_unstrat_descrip.tsv.gz") %>% 
  dplyr::select(-pathway) %>% 
  column_to_rownames("description") %>% 
  as.matrix()

picrust_pathways_matrix %<>% round(0)

picrust_pathways_matrix_Q1 <- picrust_pathways_matrix[!grepl("super", rownames(picrust_pathways_matrix)),sample_names(phy_Q1)]

#create a map data frame for pathway names

pathway_names_map_ALL <- data.frame(original_name = rownames(picrust_pathways_matrix_Q1),
                                fixed_name = make.names(rownames(picrust_pathways_matrix_Q1)))

```

## Diff abund 1: DESeq2

```{r, fig.align='center'}

dir.create("../results/Figure4/DiffAbundGenera/DESeq2/", recursive = T)

if(!file.exists("../results/Figure4/DiffAbundGenera/DESeq2/figure4_deseq2_dataframe.list.RDS")){
  
library(DESeq2)
#############################################################################

deseq_object_picrust <- phyloseq(otu_table(picrust_pathways_matrix_Q1, taxa_are_rows = TRUE), sample_data(phy_Q1))  %>% 
  core(detection = 10, prevalence = 0.05) %>% 
  phyloseq_to_deseq2(design = ~ x0_sex + x0_ager_cat + x0oh01 + smoking_exposure_ga)

geom_means <- apply(counts(deseq_object_picrust), 1, function(x) exp(mean(log(x[x>0]))))

# estimate size factors, important to account for library size variations
estim_size <- estimateSizeFactors(deseq_object_picrust, geoMeans = geom_means)

# compute the DESeq2 glm, accounting for the estimate size factor and geometric means

deseq2_results_picrust <- DESeq(estim_size, fitType = "local")

saveRDS(deseq2_results_picrust, "../results/Figure4/DiffAbundGenera/DESeq2/figure4_deseq2_dataframe.list.RDS")
} else{
  deseq2_results_picrust <- readRDS("../results/Figure4/DiffAbundGenera/DESeq2/figure4_deseq2_dataframe.list.RDS")
}

```

## Diff abund 2: LinDA

```{r, include=FALSE}
#devtools::install_github("zhouhj1994/LinDA")
dir.create("../results/Figure4/DiffAbundGenera/LinDA/")

if(!file.exists("../results/Figure4/DiffAbundGenera/LinDA/LinDA_output.list.RDS")){
  library(LinDA)
  
  LinDA_model <- linda(otu.tab = picrust_pathways_matrix_Q1, 
                   meta = phy_Q1_meta, formula = '~  x0_sex + x0_ager_cat + x0oh01 + smoking_exposure_ga', 
                   alpha = 0.05,
                   prev.cut = 0.01, 
                   lib.cut = 1000, 
                   winsor.quan = 0.97,
                   adaptive = T)
  
  saveRDS(LinDA_model, "../results/Figure4/DiffAbundGenera/LinDA/LinDA_output.list.RDS")
  }else{
    LinDA_model <- readRDS("../results/Figure4/DiffAbundGenera/LinDA/LinDA_output.list.RDS")
  }


```

## Diff abund 3: Maaslin2

```{r, include=FALSE}

dir.create("../results/Figure4/DiffAbundGenera/Maaslin2", showWarnings = FALSE)


if(!file.exists("../results/Figure4/DiffAbundGenera/Maaslin2/significant_results.tsv")){
  
  library(Maaslin2)
 # older normaliz and so on 
  # Maaslin2(
  #   input_data = picrust_pathways_matrix_Q1, 
  #   input_metadata = phy_Q1_meta, 
  #   output = "../results/Maaslin2_DA", 
  #   analysis_method = "LM",
  #   fixed_effects = c("age_cat", "sex", "x0oh01", "smoking_exposure_ga"),
  #   reference = c("smoking_exposure_ga, Never"),
  #   transform = "LOG",
  #   normalization = "TMM",
  #   correction = "BH",
  #   plot_heatmap = F
  # )

  # these following are the same settings used by Nearing  et al. (2021) 
  Maaslin2(
    input_data = picrust_pathways_matrix_Q1, 
    input_metadata = phy_Q1_meta, 
    output = "../results/Figure4/DiffAbundGenera/Maaslin2", 
    analysis_method = "LM",
    fixed_effects = c("x0_sex", "x0_ager_cat", "x0oh01", "smoking_exposure_ga"),
    reference = "smoking_exposure_ga, Never",
    transform = "LOG",
    normalization = "TSS",
    correction = "BH",
    standardize = FALSE,
    plot_heatmap = F
  )
  
}

maaslin_results <- read_tsv("../results/Figure4/DiffAbundGenera/Maaslin2/significant_results.tsv") %>% 
  mutate(feature = pathway_names_map_ALL$original_name[match(feature, pathway_names_map_ALL$fixed_name)])


```

## Diff abund 4: ALDEx2 (glm)

```{r}

dir.create("../results/Figure4/DiffAbundGenera/ALDEx2/", recursive = T)

if (!file.exists("../results/Figure4/DiffAbundGenera/ALDEx2/ALDEx2.DA.rawResults.list.RDS")) {
  
  library(ALDEx2)
  
  model_mtx <-
    model.matrix(
      ~  x0_sex + x0_ager_cat + x0oh01 + smoking_exposure_ga,
      data = dplyr::select(
        phy_Q1_meta,
        x0oh01,
        x0_sex,
        x0_ager_cat,
        smoking_exposure_ga
      ) %>% .[intersect(colnames(picrust_pathways_matrix_Q1),
                        rownames(phy_Q1_meta)), ]
    )
  
  aldex.glm.pathway_smoking_ga <-
    picrust_pathways_matrix_Q1[, intersect(colnames(picrust_pathways_matrix_Q1),
                                                     rownames(phy_Q1_meta))] %>% 
    aldex.clr(
              model_mtx,
              mc.samples = 150,
              denom = "all", 
              useMC = T
              )
  
  glm.test_pathway_smoking_ga <-
    aldex.glm(aldex.glm.pathway_smoking_ga, model_mtx)
  
  glm.effects_pathway_smoking_ga <-
    aldex.glm.effect(aldex.glm.pathway_smoking_ga, verbose = FALSE, include.sample.summary = T, useMC = T)
  
  aldex.results <- list(clr.model = aldex.glm.pathway_smoking_ga,
                        glm.test = glm.test_pathway_smoking_ga,
                        eff_size.test = glm.effects_pathway_smoking_ga)
  
  
  saveRDS(aldex.results, "../results/Figure4/DiffAbundGenera/ALDEx2/ALDEx2.DA.rawResults.list.RDS")

  detach("package:ALDEx2", unload = T)
  detach("package:zCompositions", unload = T)
  
} else{
  aldex.results <- readRDS("../results/Figure4/DiffAbundGenera/ALDEx2/ALDEx2.DA.rawResults.list.RDS")
}

```

## Diff abund 5: ANCOM-BC

```{r}

dir.create("../results/Figure4/DiffAbundGenera/ANCOM-BC/", recursive = T)

if (!file.exists("../results/Figure4/DiffAbundGenera/ANCOM-BC/ANCOM-BC.DA.rawResults.list.RDS")) {
tmp <- phyloseq(otu_table(picrust_pathways_matrix_Q1, taxa_are_rows = T), sample_data(phy_Q1_genus %>%
  select_sample_data(x0_sex, x0_ager_cat, x0oh01, smoking_exposure_ga) %>% meta())) %>%
  select_sample_data(x0_sex, x0_ager_cat, x0oh01, smoking_exposure_ga) %>% 
  core(detection = 10, prevalence = 0.05) %>%
  mia::makeTreeSummarizedExperimentFromPhyloseq()


library(ANCOMBC)

ancom_results_picrust <- ancombc2(data = tmp,
        assay_name = "counts", 
        tax_level = NULL,
        fix_formula = "x0_sex + x0_ager_cat + x0oh01 + smoking_exposure_ga", 
        group = "smoking_exposure_ga",
        struc_zero = TRUE,
        iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
        em_control = list(tol = 1e-5, max_iter = 20),  
        p_adj_method = "BH",
        n_cl = 1,
        alpha = 0.05)

saveRDS(ancom_results_picrust, "../results/Figure4/DiffAbundGenera/ANCOM-BC/ANCOM-BC.DA.rawResults.list.RDS")

}else{
  
  ancom_results_picrust <- readRDS("../results/Figure4/DiffAbundGenera/ANCOM-BC/ANCOM-BC.DA.rawResults.list.RDS")
  
}

```

# {-}

## Summary of results

Results filtered by two conditions:

  1. FDR (5%) corrected $q-value <  0.05$
  2. The effect size, irrespectively of the way it is estimated, must be larger than 0.5
  
```{r, include=FALSE}

# DESeq2
deseq2_signif_results <- list("Current/Former" = results(deseq2_results_picrust, contrast = c("smoking_exposure_ga","Current", "Former"), 
                                                          pAdjustMethod = "BH"),
                               "Current/Never" = results(deseq2_results_picrust, contrast = c("smoking_exposure_ga","Current", "Never"), 
                                                          pAdjustMethod = "BH"),
                       "Former/Never" = results(deseq2_results_picrust, contrast = c("smoking_exposure_ga","Former", "Never"), 
                                                          pAdjustMethod = "BH")
                               ) %>% 
  .$`Current/Never` %>% 
  as.data.frame() %>% 
  rownames_to_column("pathway") %>% 
  filter(padj < 0.05, abs(log2FoldChange) > 0.5)

# LinDA
LinDAresults_pathway_signif_Current <- LinDA_model$output$smoking_exposure_gaCurrent %>% 
  filter(padj < 0.05, 
         abs(log2FoldChange) > 0.5
         ) %>% 
  rownames_to_column("pathway")

# MaAsLin2
maaslin_signif <- maaslin_results %>% 
  filter(metadata == "smoking_exposure_ga") %>% 
  dplyr::select(feature, value, coef, stderr, pval, qval, N.not.0) %>% 
  filter(qval < 0.05, 
         abs(coef) > 0.5, 
         value == "Current")

# ALDEx2
aldex_gLinDAsmoking_signif <- aldex.results$glm.test %>% 
  as.data.frame() %>% 
  dplyr::select(contains("smoking_exposure_gaCurrent")) %>% 
  dplyr::rename_all(.funs = function(x) gsub(pattern =  ".*\\:", replacement = "", x = x)) %>% 
  rownames_to_column("pathway") %>% 
  as_tibble() %>% 
  filter(
    abs(Est) > 0.5, 
    pval.holm < 0.05)
  
# ANCOM-BC

ancom2_signif_results <- ancom_results_picrust$res %>% 
  dplyr::select(contains(c("pathway", "gaCurrent")), -contains("diff_")) %>% 
  rownames_to_column("pathway") %>% 
  set_names(c("pathway", "log2FoldChange", "lfcSE", "wald_stat", "p.value", "padj")) %>% 
  filter(
    abs(log2FoldChange) > 0.5,
    padj < 0.05
    )

all_models_signif_pathways <- list(ALDEx2 = aldex_gLinDAsmoking_signif %>% dplyr::select(pathway, Est) %>% set_names(c("pathway", "ALDEx2")), 
                    MaAsLin2 = maaslin_signif %>% dplyr::select(feature, coef) %>% set_names(c("pathway", "MaAsLin2")), 
                    LinDA = LinDAresults_pathway_signif_Current %>% dplyr::select(pathway,log2FoldChange) %>% set_names(c("pathway", "LinDA")), 
                    DESeq2 = deseq2_signif_results %>% dplyr::select(pathway, log2FoldChange) %>% set_names(c("pathway", "DESeq2")),
                    `ANCOM-BC` = ancom2_signif_results %>% dplyr::select(pathway, log2FoldChange) %>% set_names(c("pathway", "ANCOM-BC"))
                    ) %>% 
  purrr::reduce(., full_join, by = "pathway")



frequency_signif_pathway <- pivot_longer(all_models_signif_pathways, cols = !starts_with("pathway"), names_to = "DA_method", values_to = "da_eff_size") %>%
  filter(!is.na(da_eff_size)) %>% 
  group_by(pathway) %>% 
  tally() %>% 
  arrange(desc(n))

pathways_union <- frequency_signif_pathway$pathway

pathways_consensus <- frequency_signif_pathway %>% 
  filter(n >= 4) %>% 
  .$pathway

pathways_intersect <- frequency_signif_pathway %>% 
  filter(n == 5) %>% 
  .$pathway


saveRDS(pathways_union, "../results/Figure4/significant_PWYs_union.RDS")
saveRDS(pathways_intersect, "../results/Figure4/significant_PWYs_intersect.RDS")
saveRDS(pathways_consensus,  "../results/Figure4/significant_PWYs_consensus.RDS")
```

  * Relative Abundance
  * Mean per smoking group
  * Z-scaled for plotting
  
```{r, fig.height=9, fig.width=11}
if(!exists("pathways_consensus")){
  pathways_consensus <- readRDS("../results/Figure4/significant_PWYs_consensus.RDS")
}

dummy_tax_table <- data.frame(pathway = rownames(picrust_pathways_matrix_Q1)) %>% set_rownames(rownames(picrust_pathways_matrix_Q1)) %>% as.matrix()


pathways_summarized_plot <- phyloseq(otu_table(picrust_pathways_matrix_Q1, taxa_are_rows = T),
         sample_data(phy_Q1_meta), tax_table(dummy_tax_table)) %>% 
  phy_transform("compositional") %>% 
  filter_sample_data(sample_name %in% c(sample_names(phy_Q2), sample_names(phy_Q3)) | smoking_exposure_ga == "Never") %>% 
  filter_tax_table(taxa_names(.) %in% pathways_consensus) %>% 
  mutate_sample_data(smoking_detailed = add_N_to_catVar(smoking_detailed)) %>%
  phy_summarise_taxa_by_metadata(metadata_var = "smoking_detailed", .fun = mean)

ggsave(plot = ggplotify::as.ggplot(pheatmap::pheatmap(pathways_summarized_plot, scale = "none", main = "Unscaled Rows", cluster_cols = FALSE, angle_col = 315)), 
       filename = "../results/Figure4/Figure 4 picrust_unscaledRows.png", dpi = 600, height = 6, width = 8)

# save also scaled rows for potential alternative use

ggsave(plot = ggplotify::as.ggplot(pheatmap::pheatmap(pathways_summarized_plot, scale = "row", main = "Scaled Rows", cluster_cols = FALSE, angle_col = 315)), 
       filename = "../results/Figure4/Figure4 picrust_scaledRows.png", dpi = 600, height = 6, width = 8)


```

Interesting note, the CLR transformation is dependent somehow on the number of features in the table, meaning that if you subset before creating the phyloseq object to make the transformation, you end up with a correct transformation, but scaled only on those pathways you selected. In my case, this approach was masking a few signals, so my advice is: 
  
  - create a phyloseq object with all features (taxa, pathways, Kegg Orthology enzymes ...) (optional, I use it because the transformaton functions are implemented)
  - re-convert into `data.frame` with either `speedyseq::psmelt()` or `phy_OtuMetaTable()`
  - make all the plots you want

Notice that many pathways are actually highly correlated, like the quinols. let's see this in a heatmap

```{r,fig.height=7, fig.width=11}

signif_pathways_names <- readRDS("../results/Figure4/significant_PWYs_consensus.RDS")

picrust_pathways_matrix_Q1[signif_pathways_names,] %>% 
  apply(2, function(s) s/sum(s)) %>% 
  t() %>% 
  stats::cor(method = "spearman") %>% 
  pheatmap::pheatmap(show_colnames = FALSE, main = "Spearman correlation between significant pathways")

```

```{r, include=FALSE}

NA_value <- -5

heatmap_plot <- ggplotify::as.ggplot(all_models_signif_pathways %>% 
  column_to_rownames("pathway") %>% 
  mutate_all(.funs = function(x) ifelse(is.na(x), NA_value, x)) %>% 
  as.matrix() %>% 
  pheatmap::pheatmap(angle_col = 315, treeheight_row = 0, treeheight_col = 9, scale = "none")
)

plot_correlations_eff.sizes <- grid::grid.grabExpr(
  print(all_models_signif_pathways %>% 
          filter(complete.cases(.)) %>% 
  column_to_rownames("pathway") %>% 
 mutate_all(.funs = function(x) ifelse(is.na(x), 0, x)) %>%
 #as.matrix() %>%
  GGally::ggpairs(title = "Correlations between effect sizes\nof the features identified in all methods")))

library(ggVennDiagram)

datavenn <- list(ALDEx2 = aldex_gLinDAsmoking_signif %>% dplyr::select(pathway) %>% set_names(c("pathway", "ALDEx2")), 
                               MaAsLin2 = maaslin_signif %>% dplyr::select(feature, coef) %>% set_names(c("pathway", "MaAsLin2")), 
                               LinDA = LinDAresults_pathway_signif_Current %>%  dplyr::select(pathway,log2FoldChange) %>% set_names(c("pathway", "LinDA")), 
                               DESeq2 = deseq2_signif_results %>% dplyr::select(pathway, log2FoldChange) %>% set_names(c("pathway", "DESeq2"))
                               ) %>% lapply("[[", "pathway") %>% 
  Venn() %>% 
  process_data()


venn_plot <- ggplot()+
  geom_sf(aes(fill = count), data = venn_region(datavenn)) + 
  scale_fill_gradient2(
    low = "white",
    high = scales::muted("red")
  )+
  geom_sf(color="darkgrey", size = 2, data = ggVennDiagram::venn_setedge(datavenn))+
  geom_sf_text(aes(label = name), color = "black", fontface = "bold", data = venn_setlabel(datavenn)) +  
  # add a alternative region name
  geom_sf_label(aes(label = paste0(count, "\n", round(count/sum(count)*100, 1), "%")), 
                data = venn_region(datavenn), 
                alpha = 0.5, label.size = NA) +  
  theme_void()+
  theme(legend.position = "none")

```

```{r, fig.height=10, fig.width=12}

ggarrange(ggarrange(plot_correlations_eff.sizes, 
                    venn_plot+
                      scale_x_continuous(expand = expansion(mult = .2)), 
                    
                    nrow = 2, ncol = 1, heights = c(1,0.5)
                    ), 
          
          heatmap_plot, 
          
          nrow = 1, ncol = 2
          ) %>% 
  annotate_figure(top = text_grob("Discussion of differential abundance\nresults similarity across methods", size = 16, face = "bold", color = "red2"),
                  bottom = text_grob(paste("NAs in the heatmap were arbitrarily set to", NA_value, "in order not to fail clustering,\nwhile still keeping evident track of where a method did not return a pathway as significant"), size = 10))

```

From these results we find out that effect sizes and the number o ALDEx2 and ANCOM-BC are further away, and that LinDA, despite being considered by authors as very similar to ANCOM-BC, in our results it seems more similar to MaAsLin2, with the added value of being extremely fast. 
