---
title: "Smoking Paper - Appendix A" 
subtitle: "Genus level"
author: "Giacomo Antonello"
date: "`r Sys.Date()`"
output: word_document
always_allow_html: true
---

```{r, echo=FALSE}

knitr::opts_chunk$set(
  cache = FALSE,
  concordance = TRUE,
  prompt = TRUE, # fig.width=5, fig.height=5,
  out.width = "100%",
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  error = TRUE,
  tidy = FALSE,
  comment = ""
  )

```

# Data preparation {.tabset}

```{r library setup}
# from Bioconductor (install by running: BiocManager::install("x") | x = name of the package)
# BiocManager::install(c("DESeq2", "phyloseq", "microbiome"))


packages_github_links <- list(
  speedyseq = "mikemc/speedyseq",
  gautils = "g-antonello/gautils",
  LinDA = "zhouhj1994/LinDA"
)
missing_github_pkgs <- !sapply(names(packages_github_links), function(pkg) pkg %in% rownames(installed.packages()))
missing_github_pkgs <- names(missing_github_pkgs[missing_github_pkgs])

if(length(missing_github_pkgs) > 0){
sapply(missing_github_pkgs, function(pkg) devtools::install_github(packages_github_links[[pkg]]))
}

library(gautils)
library(magrittr) # imported separately from tidyverse to also import '%<>%'
library(ggpubr)
library(kableExtra)


# graphics packages
# library(ggpubr)
# library(magrittr)
# # my functions
# source("my_personal_functions.R")
# 
# 
# library(speedyseq)
theme_set(theme_light())


chrismb_phy <- readRDS("/shared/statgen/microbiome/CHRISMB/processed_data/microbiome_tables/phyloseq_built.Rds")


threshold_prevalence <- 0.01
threshold_detection <- 10


```

## Download CHRIS data

These are based on data retrieval from the `chrisData` internal R package curated by J. Rainer. Variable selection is done via consulting the codebook. 

```{r, message=FALSE, warning=FALSE}
chrismb_phy <- readRDS("/shared/statgen/microbiome/CHRISMB/processed_data/microbiome_tables/phyloseq_built.Rds")

library(chrisData)

possible_data_files <- chrisDataFile()

selected_data_files <- c(
    "general_information",
    "household",
    "clinical_traits",
    "interview",
    "labdata",
    "substudies",
    "drugs_summary"
  )

data_selected.list <- sapply(selected_data_files, function(x) chrisData(paste0(x, ".txt.gz")), USE.NAMES = T)

data_selected.joined.df <- purrr::reduce(data_selected.list, full_join, by = "AID")
data_selected.joined.df <- data_selected.joined.df[match(chrismb_phy@sam_data$AID, data_selected.joined.df$AID),]

# Columns with all NAs must be necessarily removed 
all_NAs <- apply(data_selected.joined.df, 2, function(x) all(is.na(x)))
data_selected.joined.df <- data_selected.joined.df[, !all_NAs]

# Columns of class "chron" are problematic, they must be removed to be able to work with phyloseq and metadata data.frame back and forth
data_selected.joined_NODates.df <- select(data_selected.joined.df, -contains(c("x0bp08", "x0bc03", "x0bc06")))

rm(data_selected.list)

chrismb_phy <- phy_add_metadata_variables(chrismb_phy, df = data_selected.joined_NODates.df, by = "AID")


levels(chrismb_phy@sam_data$x0oh01) <- c("0", "1-9", "10-19", "20+")
levels(chrismb_phy@sam_data$x0sm51) <- c("Never", "Former", "Current (R)", "Current (NR)")
chrismb_phy@sam_data$x0an03a %<>% as.numeric()

chrismb_phy@sam_data$x0_ager_cat <- cut(chrismb_phy@sam_data$x0_ager, breaks = c(18, 31, 41, 51, 61, 71, max(chrismb_phy@sam_data$x0_ager)+1), include.lowest = TRUE, labels = c("18-30", "31-40", "41-50", "51-60", "61-70", "71+"))

# create some househod (hid) info
chrismb_phy <- relocate_sample_data(chrismb_phy, x0_ager_cat, .after = x0_ager) %>% 
  mutate_sample_data(
    hid = paste0("h", hid),
    hid_size_chrismb = add_N_to_catVar(hid) %>% as.character() %>% gsub("h.*\\ ", "", .) %>% parse_number(),
  smoking_exposure_ga = 
    case_when(
      grepl("Current", as.character(x0sm51)) ~ "Current",
      TRUE ~ as.character(x0sm51)
      ) %>% 
    factor(levels = c("Never", "Former", "Current"))
  ) %>% 
  relocate_sample_data(smoking_exposure_ga, .before = x0sm51)

```

## Smoking variables generation

### Bin smoking intensity (g/day)

```{r}

###############################
## bin daily tobacco smoked
###############################

chrismb_phy %<>%
  mutate_sample_data(
    ### tobacco g per day (bin 5 by 5)
    x0sm61_bin5cat = case_when(smoking_exposure_ga == "Current" & !is.na(x0sm61) ~ cut(
      x0sm61,
      breaks = c(0, 1, 2, seq(5, 30, 5), 61),
      include.lowest = TRUE
    )),
  ### tobacco g per day (bin 5 by 5, semi-continuous, using upper limit of the range as integer)
  x0sm61_bin5cont = case_when(smoking_exposure_ga == "Current" & !is.na(x0sm61) ~ cut(
      x0sm61,
      breaks = c(0, 1, 2, seq(5, 30, 5), 60),
      labels = c(1, 2, seq(5, 30, 5), 60),
      include.lowest = TRUE
    )) %>% as.character() %>% as.numeric(),
  
    ### tobacco g per day (bin 2 by 2)
    x0sm61_bin2cat = case_when(smoking_exposure_ga == "Current" & !is.na(x0sm61) ~ cut(
      x0sm61,
      breaks = c(0, 1, seq(2, 30, 2), 60),
      include.lowest = TRUE
    )),
  
  ### tobacco g per day (bin 2 by 2, semi-continuous, using upper limit of the range as integer)
  x0sm61_bin2cont = case_when(smoking_exposure_ga == "Current" & !is.na(x0sm61) ~ cut(
      x0sm61,
      breaks = c(0, 1, seq(2, 30, 2), 60),
      labels = c(1, seq(2, 30, 2), 60),
      include.lowest = TRUE
    )
    ) %>% as.character() %>% as.numeric()
  )

```

### Bin Years since quitting

```{r}
# x0sm40 = age when people quit or reduce
# x0_ager = age rounded to closest integer

max_value <- max(chrismb_phy@sam_data$x0_ager - chrismb_phy@sam_data$x0sm40, na.rm = T)

chrismb_phy %<>% 
  mutate_sample_data(
    years_since_quitting_or_reducing = x0_ager - x0sm40,
    
    years_since_quitting_binned = case_when(
      smoking_exposure_ga == "Former" & !is.na(years_since_quitting_or_reducing) ~ cut(years_since_quitting_or_reducing, 
                                            breaks = c(0,1,3,5,10,max_value), 
      include.lowest = TRUE
    )
  )
)

rm(max_value)

```

### Ultimate smoking variable for plots: "smoking_detailed"

```{r}

chrismb_phy %<>% 
  mutate_sample_data(
    smoking_detailed = case_when(
      smoking_exposure_ga == "Never" ~ "Never",
      
      smoking_exposure_ga == "Former" ~ paste(smoking_exposure_ga,
                                               years_since_quitting_binned, 
                                               "y"),
      smoking_exposure_ga == "Current" ~ paste(smoking_exposure_ga,
                                               x0sm61_bin5cat,
                                               "g"),
      TRUE ~ "Other"
    ),
    smoking_detailed.Extra = case_when(
      smoking_exposure_ga == "Never" ~ "Never",
      
      smoking_exposure_ga == "Former" ~ paste(smoking_exposure_ga,
                                               years_since_quitting_or_reducing, 
                                               "y"),
      smoking_exposure_ga == "Current" ~ paste(smoking_exposure_ga,
                                               x0sm61,
                                               "g"),
      TRUE ~ "Other"
    )
    
  )



nice_order_current <- sort(grep("Current", unique(chrismb_phy@sam_data$smoking_detailed), value = TRUE))[c(9,1,4,8,2,3,5,6,7,10)] %>% rev()

nice_order_former <- sort(grep("Former", unique(chrismb_phy@sam_data$smoking_detailed), value = TRUE))[c(5,1,3,4,2,6)]

final_levels <- c(nice_order_current,nice_order_former, "Never")


# SMOKING DETAILED 1 - make ordered factor, useful for plotting
chrismb_phy@sam_data$smoking_detailed <- factor(chrismb_phy@sam_data$smoking_detailed, levels = final_levels)


```

## Available Samples for statistical analysis

```{r}
excluded_samples_upon_request <- readLines("/home/gantonello/CHRISMB/participants lists each study/withdrawnConsents_AIDs.txt")

chrismb_phy_core <- core(chrismb_phy, detection = threshold_detection, prevalence = threshold_prevalence)

phy_Q1 <- chrismb_phy_core %>% 
  filter_sample_data(!(AID %in% excluded_samples_upon_request),
                     !is.na(x0oh01),
                     !is.na(x0_ager),
                     !is.na(x0_sex),
                     !is.na(x0sm51),
                     #x0bc10 != 1, # people who smoked were reported as 1, NAs are treated as non-smoking
                     x0dd51 == "No") 

phy_Q1


phy_Q1_genus <- phy_tax_glom(phy_Q1, "Genus")


phy_Q1_meta <- meta(phy_Q1)

```

## Generate directories to save data and figures

```{r, include=F}

sapply(paste0("A", 1:9), function(x) dir.create(paste0("appendix_results/",x), showWarnings = F, recursive = T))

```

# Supplementary Figures

## Figure A1 - Differential Abundant genera in comparable literature

```{r}

# load significant genera found in Figure 1 and average the effect sizes, then get the sign (-1, +1)
diff_abundant_genera_CHRISMB <- readRDS("../Main/results/Figure1/DiffAbundGenera/significant_genera_Consensus-and-log2FC.RDS") %>% 
  filter(taxon %in% readRDS("../Main/results/Figure1/DiffAbundGenera/significant_genera_consensus.RDS")) %>% 
  column_to_rownames("taxon") %>% 
  rowMeans(na.rm = T) %>% 
  as.data.frame() %>% 
  rownames_to_column("Genus") %>% 
  set_names(c("Genus", "CHRISMB")) %>% 
  mutate(CHRISMB = sign(CHRISMB))

# load table of genera found in the literature
excel_map <- read_tsv("appendix-input-data/diff_abund_genera_matrix_for_plots.tsv") %>% 
  full_join(diff_abundant_genera_CHRISMB, by = "Genus") %>% 
  column_to_rownames("Genus") %>% 
  as.matrix()

excel_map[is.na(excel_map)] <- 0

```

```{r}

excel_map_no_rare_genera <- excel_map[rowSums(abs(excel_map)) >= 3,]

clust_cols <- hclust(dist(t(
  excel_map_no_rare_genera
)))
columns_ordered <- clust_cols$labels[clust_cols$order]

clust_rows <- hclust(dist(excel_map_no_rare_genera))
rows_ordered <- clust_rows$labels[clust_rows$order] %>% rev()

heatmap_ggplot_common <- reshape2::melt(excel_map_no_rare_genera)  %>% 
  mutate(Var1 = factor(Var1, levels = rows_ordered), 
         Var2 = factor(Var2, levels = columns_ordered)) %>% 
  ggplot(aes(x = Var2, y = Var1, fill = factor(value)))+ 
  geom_tile(color="darkgray")+
  scale_fill_manual(name = "Differential Abundance", 
                    values = c("#2C54C7", "#FBF2BC", "#C72E2C"), 
                    labels = c("Decreased", "NA/NS", "Increased"))+ 
  theme(
    panel.border = element_blank(), 
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = -45, hjust = 0),
    axis.text.y = element_text(hjust = 1)
    )+
  labs(
    y = "Genus",
    x = NULL
  )

```

```{r}

excel_map_genera_only1 <- excel_map[rowSums(abs(excel_map)) == 1,]
excel_map_genera_only1 <- excel_map_genera_only1[, colSums(abs(excel_map_genera_only1)) > 0]

clust_cols <- hclust(dist(t(
  excel_map_genera_only1
)))
columns_ordered <- clust_cols$labels[clust_cols$order]

clust_rows <- hclust(dist(excel_map_genera_only1))
rows_ordered <- clust_rows$labels[clust_rows$order] %>% rev()

heatmap_ggplot_rare <- reshape2::melt(excel_map_genera_only1)  %>% 
  mutate(Var1 = factor(Var1, levels = rows_ordered), 
         Var2 = factor(Var2, levels = columns_ordered)) %>% 
  ggplot(aes(x = Var2, y = Var1, fill = factor(value)))+ 
  geom_tile(color="darkgray")+
  scale_fill_manual(name = "Differential Abundance Direction", 
                    values = c("#2C54C7", "#FBF2BC", "#C72E2C"), 
                    labels = c("Decreased", "NA/NS", "Increased"))+
  theme(
    panel.border = element_blank(), 
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = -45, hjust = 0),
    axis.text.y = element_text(hjust = 1)
    )+
  labs(
    y = "Genus",
    x = NULL
  )

```


**Figure A1. Heatmap of differentially abundant genera in smokers compared to non-smokers both in the literature and in the current study (CHRISMB).** Genera were manually annotated based on publications with comparable setup: cross-sectional, on a healthy population, and 16S amplicon microbiota data. The red color means increased abundance in smokers, the blue color means decreased abundance in smokers. The yellow color means not significant (NS) or not mentioned (NA) in the text nor in the supplementary materials. **(A)** Commonly reported genera (present in at least 3 studies). **(B)** Rarely reported genera (present in only one study).

```{r, fig.width=12, fig.height=9}

A1_plot <- ggarrange(heatmap_ggplot_common, heatmap_ggplot_rare+ labs(y = NULL), common.legend = T)

ggsave(plot = A1_plot, 
       filename = "appendix_results/A1/A1_heatmap_diff_abund.genera.png", 
       height = 9, 
       width = 12, 
       dpi = 300)

ggarrange(heatmap_ggplot_common, heatmap_ggplot_rare+ labs(y = NULL), common.legend = T)
```

## Figure A2 - histogram of grams per day in current smokers

**Figure A2. Histogram of grams of tobacco smoked daily by CHRISMB current smokers (N = 309)**

```{r}
A2_plot <- phy_Q1_meta %>% 
  filter(smoking_exposure_ga == "Current") %>% 
  ggplot(aes(x = x0sm61))+
  geom_histogram()+
  labs(
    y = "Counts",
    x = "Grams of tobacco smoked per day",
    caption = "Tobacco is an integration of cigarettes, cigars, cigarillos and pipe.\nHowever, the vast majority of smokers smoke cigarettes exclusively"
  )

ggsave(
  plot = A2_plot, 
  filename = "appendix_results/A2/A2_histogram of smoking intensity values.png", 
  height = 5,
  width = 7,
  dpi = 300
)

A2_plot

```


## Figure A3 - Age, Teeth and Smoking

**Figure A3: Non-independence of number of teeth and age categories in the effective CHRISMB dataset for the smoking analysis.** (A) Relationship between age and number of teeth. Numbers are reported as proportions for each age category. (B) Relationship between lifetime exposure to smoke (lifetime pack-years) and the number of teeth in only current smokers. Individuals were further split into age groups to control for age-related tooth loss. Lifetime pack years were calculated in Former and Current smokers as follows: 

$$
Lifetime~Pack\text-Year~equivalents = \frac{\frac{cigarettes}{day}}{20} \times 365~days \times years~smoked
$$


```{r}

Fig_A3.A <- phy_Q1_meta %>% 
  group_by(x0_ager_cat, x0oh01) %>% 
  tally() %>% 
  ungroup() %>% 
  group_by(x0_ager_cat) %>%
  mutate(n = n/sum(n)*100) %>% 
  
  ggplot(aes(y = x0_ager_cat, x = n, fill = x0oh01))+
  geom_bar(stat= "identity")+
  scale_fill_viridis_d()+
  labs(x = "Proportion of Individuals (%)",
       y = "Age Category",
       fill = "Number of Teeth")+
  theme(legend.position = "top")

```

```{r}
Fig.A3B <- phy_Q1_meta %>% 
  filter(smoking_exposure_ga != "Current") %>% 
  mutate(x0_ager_cat = paste(x0_ager_cat, "years")) %>% 
  ggplot(aes(x = x0oh01, y = x0sm55))+ 
  geom_boxplot()+facet_wrap(~x0_ager_cat)+
  EnvStats::stat_n_text()+ 
  labs(x = "Number of Teeth", 
       y = "Lifetime Pack-Year equivalents"
  )

```

```{r, fig.width=7, fig.height=8}

age_teeth_smoking <- suppressWarnings(ggpubr::ggarrange(Fig_A3.A, Fig.A3B, ncol = 1, nrow = 2, labels = "AUTO", heights = c(0.6,1)))

ggsave(plot = age_teeth_smoking, 
       filename = "appendix_results/A3/A3_teeth_vs_smoking.png", 
       width = 7, 
       height = 9, 
       dpi = 300)

age_teeth_smoking
```


## Figure A4 - Current (NR) are indistinguishable from Current (R) {.tabset}

Looking at the figure below, we can appreciate a mild change in mean composition (Figure A3, B), but not a major one, also supported by absence of separation of individuals in the `Current (R)` and `Current (NR)` groups.

```{r, fig.height=12, fig.width=12, include=FALSE}

phy_Q1_A3 <- phy_tax_glom(phy_Q1, level = "Genus") %>% 
  filter_sample_data(smoking_exposure_ga == "Current") %>% 
  mutate_sample_data(x0sm51 = droplevels(x0sm51))

div <- phyloseq::distance(microbiome::transform(phy_Q1_A3, "compositional"), "bray")

ord <- ordinate(physeq = phy_Q1_A3,
                distance = div,
                method = "PCoA"
)

fig1A <- plot_ordination(physeq = phy_Q1_A3,
                         ordination = ord, 
                         color = "x0sm51")+ 
  labs(title = NULL)+
  theme_light()+
  theme(legend.position = "bottom")+ 
  scale_color_discrete(name = "Smoking habit",type = ggsci::pal_jco("default")(2))+ 
  stat_ellipse(aes(group = x0sm51), show.legend = FALSE)+
  guides(color = guide_legend(override.aes = list(size=3)))

fig1A$labels$caption <- gsub(fig1A$labels$caption, pattern = "\n", replacement = "; ")
fig1A$labels$x <- paste0("Principal Coordinate 1   ", strsplit(fig1A$labels$x, "   ")[[1]][[2]])
fig1A$labels$y <- paste0("Principal Coordinate 2   ", strsplit(fig1A$labels$y, "   ")[[1]][[2]])

###################################################################################
# Fig 1 B
# oxygen 3 plots
library(rstatix)
df_final <- readRDS("../Main/results/Figure1/OxygenMetabolism/oxygen_with_metadata_df.RDS")

stat_to_plot <- filter(df_final, oxygen != "NA", smoking_exposure_ga == "Current")%>%
  group_by(oxygen) %>%
  rstatix::wilcox_test(value ~ x0sm51) %>%
  adjust_pvalue(method = "BH") %>%
  #add_significance("p.adj") %>%
  add_xy_position(x = "oxygen", step.increase = 0.05) # this step takes the longest

fig1C <- ggboxplot(data = filter(df_final, oxygen != "NA", smoking_exposure_ga == "Current"),
                   x = "oxygen", 
                   y = "value", 
                   fill = "x0sm51") + 
  ylab("") + 
  xlab("")+ 
  scale_fill_discrete(name = "Smoking habit", 
                      type = ggsci::pal_jco()(4)) + 
  theme(legend.position = "top") +
  stat_pvalue_manual(stat_to_plot, label =  "p.adj", tip.length = 0.01) + 
  labs(caption = "Benjamini-Hochberg FDR (5%) adjusted Wilcoxon test q-values")+ 
  theme_light()+
  theme(legend.position = "none")


###################################################################################
```

### Diff abund 1: DESeq2

```{r, fig.align='center'}

library(DESeq2)
#############################################################################

deseq_object_genus <-
  phyloseq_to_deseq2(phy_Q1_A3, design = ~ x0_sex + x0_ager_cat + x0oh01 + x0sm51)

geom_means <- apply(counts(deseq_object_genus), 1, function(x) exp(mean(log(x[x>0]))))

# estimate size factors, important to account for library size variations
estim_size <- estimateSizeFactors(deseq_object_genus, geoMeans = geom_means)

# compute the DESeq2 glm, accounting for the estimate size factor and geometric means

deseq2_results_genus <- DESeq(estim_size, fitType = "local")

figA3_diffAbund_DESeq2 <- results(deseq2_results_genus, c("x0sm51", "Current (R)", "Current (NR)")) %>% 
  as.data.frame() %>% 
  arrange(padj)

```

### Diff abund 2: LinDA

```{r, include=FALSE}
#devtools::install_github("zhouhj1994/LinDA")
library(LinDA)

LinDA_model_genera <- linda(otu.tab = abundances(phy_Q1_A3),
                   meta = meta(phy_Q1_A3), 
                   formula =  "~ x0_sex + x0_ager_cat + x0oh01 + x0sm51", 
                   alpha = 0.05,
                   prev.cut = 0.01, 
                   lib.cut = 1000, 
                   winsor.quan = 0.97,
                   adaptive = T)

```

### Diff abund 3: Maaslin2

```{r, include=FALSE}

dir.create("appendix_results/A3/Maaslin2_DA/", showWarnings = FALSE, recursive = T)


if(!file.exists("appendix_results/A3/Maaslin2_DA/significant_results.tsv")){
  
  library(Maaslin2)
  
 # older normaliz and so on 
  # Maaslin2(
  #   input_data = picrust_taxa_matrix_Q1, 
  #   input_metadata = phy_Q1_meta, 
  #   output = "appendix_results/Maaslin2_DA", 
  #   analysis_method = "LM",
  #   fixed_effects = c("age_cat", "sex", "x0oh01", "x0sm51"),
  #   reference = c("x0sm51, Never"),
  #   transform = "LOG",
  #   normalization = "TMM",
  #   correction = "BH",
  #   plot_heatmap = F
  # )

  # these following are the same settings used by Nearing  et al. (2021) 
  capture.output(Maaslin2(
    input_data = abundances(phy_Q1_A3), 
    input_metadata = meta(phy_Q1_A3), 
    output = "appendix_results/A3/Maaslin2_DA/", 
    analysis_method = "LM",
    fixed_effects = c("x0_sex", "x0_ager_cat", "x0oh01","x0sm51"),
    reference = "x0sm51, Never",
    transform = "LOG",
    normalization = "TSS",
    correction = "BH",
    standardize = FALSE,
    plot_heatmap = F,
    plot_scatter = F
  ), file = "appendix_results/A3/Maaslin2_DA/log.txt"
  )

}

genera_names_map_all <- as.data.frame(tax_table(phy_Q1_A3))["Genus"] %>% 
  mutate(feature = make.names(Genus))

maaslin_results_genera <- read_tsv("appendix_results/A3/Maaslin2_DA/significant_results.tsv") %>% 
  left_join(genera_names_map_all, by = "feature") %>% 
  mutate(feature = Genus) %>% 
  dplyr::select(-Genus)


```

### Diff abund 4: ALDEx2 (glm)

```{r}

library(ALDEx2)

# define model matrix, based on the metadata
model_mtx <-
  model.matrix(
    ~ x0_sex + x0_ager_cat + x0oh01 + x0sm51,
    data = dplyr::select(meta(phy_Q1_A3),
                         x0oh01,
                         x0_sex,
                         x0_ager_cat,
                         x0sm51)
  )
# perform the clr differential abundance
aldex.glm.genera_smoking_ga <-
  aldex.clr(
    abundances(phy_Q1_A3),
    model_mtx,
    mc.samples = 200,
    denom = "all",
    useMC = F
  )

# make glm, obtain significance of terms
glm.test_genera_smoking_ga <-
  aldex.glm(aldex.glm.genera_smoking_ga, model_mtx)

```

### Diff abund 5: ANCOM-BC

```{r}

tmp <- mia::makeTreeSummarizedExperimentFromPhyloseq(phy_Q1_A3 %>% select_sample_data(x0_sex, x0_ager_cat, x0oh01, x0sm51)) 


library(ANCOMBC)

# takes around 30 minutes
ancom_results <- ancombc2(data = tmp,
        assay_name = "counts", 
        tax_level = NULL,
        fix_formula = "x0_sex + x0_ager_cat + x0oh01 + x0sm51", 
        group = "x0sm51",
        struc_zero = TRUE,
        iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
        em_control = list(tol = 1e-5, max_iter = 20),  
        p_adj_method = "BH",
        n_cl = 1,
        alpha = 0.05)

```

## {-}

### Summary of results

Results filtered by two conditions:

  1. FDR (5%) corrected $q-value < 0.05$
  2. The effect size, irrespectively of the way it is estimated, must be larger than 0.5

Since this value is rather arbitrary, one could devise a *quantile cutoff*, but it's hard to decide where to cut, since we have no *a priori* expectations of the effect size distributions.
  
```{r, include=FALSE}

library(DESeq2)

# DESeq2
deseq2_signif_results <- figA3_diffAbund_DESeq2 %>% 
  as.data.frame() %>% 
  rownames_to_column("taxon") %>% 
  filter(padj < 0.05)

# LinDA
LinDAresults_taxon_signif_Current <- LinDA_model_genera$output$`x0sm51Current (NR)` %>% 
  filter(padj < 0.05, 
         #abs(log2FoldChange) > 0.5
         ) %>% 
  rownames_to_column("taxon")

# MaAsLin2
maaslin_signif <- maaslin_results_genera %>% 
  filter(metadata == "x0sm51") %>% 
  dplyr::select(feature, value, coef, stderr, pval, qval, N.not.0) %>% 
  filter(qval < 0.05, 
         #abs(coef) > 0.5, 
         value == "Current")

# ALDEx2s
aldex_glm_smoking_signif <- glm.test_genera_smoking_ga %>% 
  as.data.frame() %>% 
  dplyr::select(contains("x0sm51Current")) %>% 
  dplyr::rename_all(.funs = function(x) gsub(pattern =  ".*\\:", replacement = "", x = x)) %>% 
  rownames_to_column("taxon") %>% 
  as_tibble() %>% 
  filter(
    #abs(Est) > 0.5, 
    pval.holm < 0.05)
  
# ANCOM-BC

ancom2_signif_results <- ancom_results$res %>% 
  filter(
    #abs(log2FoldChange) > 0.5,
    `q_x0sm51Current (NR)` < 0.05
    )

tableDA.Taxa_A3 <- cbind(c("DESeq2", "LinDA", "MaAsLin2", "ALDEx2", "ANCOM-BC"),
                         c(nrow(deseq2_signif_results), nrow(LinDAresults_taxon_signif_Current), nrow(maaslin_signif), nrow(aldex_glm_smoking_signif), nrow(ancom2_signif_results))) %>% 
  set_colnames(c("Diff. Abund Method", "N significant Taxa"))

```


```{r, fig.height=14, fig.width=10, fig.cap = "Current Smokers who reduced (NR) are not different than current smokers who reduced (R). Left: Table of the differentially abundant genera between Current (NR) and Current (R)."}

A4_plot <- ggarrange(ggtexttable(tableDA.Taxa_A3), ggarrange(fig1A, fig1C+ labs(caption = NULL) + ylim(c(0, 1)), nrow = 2, ncol = 1, common.legend = T), ncol = 2, nrow = 2)

ggsave(plot = A4_plot, 
       filename = "appendix_results/A4/A4_smokers_indistinguishable.png", 
       width = 8, 
       height = 12, 
       dpi = 300)

A4_plot
```


## Figure A5 - Differential abundance split by sex

### Males {.tabset}

```{r}
phy_Q1_A4 <-filter_sample_data(phy_Q1_genus, x0_sex == "Male")
```

#### Diff abund 1: DESeq2

```{r, fig.align='center'}

dir.create("appendix_results/A4/DiffAbund_Males/DESeq2/", recursive = T)

if(!file.exists("appendix_results/A4/DiffAbund_Males/DESeq2/figureA4_deseq2_dataframe.list.RDS")){
library(DESeq2)
#############################################################################

deseq_object_genus <-
  phyloseq_to_deseq2(phy_Q1_A4, design = ~ x0_ager_cat + x0oh01 + smoking_exposure_ga)

geom_means <- apply(counts(deseq_object_genus), 1, function(x) exp(mean(log(x[x>0]))))

# estimate size factors, important to account for library size variations
estim_size <- estimateSizeFactors(deseq_object_genus, geoMeans = geom_means)

# compute the DESeq2 glm, accounting for the estimate size factor and geometric means

deseq2_results_genus <- DESeq(estim_size, fitType = "local")


saveRDS(deseq2_results_genus, "appendix_results/A4/DiffAbund_Males/DESeq2/figureA4_deseq2_dataframe.list.RDS")
} else{
  deseq2_results_genus <- readRDS("appendix_results/A4/DiffAbund_Males/DESeq2/figureA4_deseq2_dataframe.list.RDS")
}

```

#### Diff abund 2: LinDA

```{r, include=FALSE}
#devtools::install_github("zhouhj1994/LinDA")

dir.create("appendix_results/A4/DiffAbund_Males/LinDA/", recursive = T)

if(!file.exists("appendix_results/A4/DiffAbund_Males/LinDA/LinDA_output.list.RDS")){
  
library(LinDA)

LinDA_model_genera <- linda(otu.tab = abundances(phy_Q1_A4),
                   meta = meta(phy_Q1_A4), 
                   formula =  "~ x0_ager_cat + x0oh01 + smoking_exposure_ga", 
                   alpha = 0.05,
                   prev.cut = 0.01, 
                   lib.cut = 1000, 
                   winsor.quan = 0.97,
                   adaptive = T)

saveRDS(LinDA_model_genera, "appendix_results/A4/DiffAbund_Males/LinDA/LinDA_output.list.RDS")
}else{
  LinDA_model_genera <- readRDS("appendix_results/A4/DiffAbund_Males/LinDA/LinDA_output.list.RDS")
}

```


#### Diff abund 3: Maaslin2

```{r, include=FALSE}

dir.create("appendix_results/A4/DiffAbund_Males/Maaslin2", showWarnings = FALSE, recursive = T)


if(!file.exists("appendix_results/A4/DiffAbund_Males/Maaslin2/significant_results.tsv")){
  
  library(Maaslin2)
 # older normaliz and so on 
  # Maaslin2(
  #   input_data = picrust_taxa_matrix_Q1, 
  #   input_metadata = phy_Q1_meta, 
  #   output = "../results/Maaslin2_DA", 
  #   analysis_method = "LM",
  #   fixed_effects = c("age_cat", "sex", "x0oh01", "smoking_exposure_ga"),
  #   reference = c("smoking_exposure_ga, Never"),
  #   transform = "LOG",
  #   normalization = "TMM",
  #   correction = "BH",
  #   plot_heatmap = F
  # )

  # these following are the same settings used by Nearing  et al. (2021) 
  Maaslin2(
    input_data = abundances(phy_Q1_A4), 
    input_metadata = meta(phy_Q1_A4), 
    output = "appendix_results/A4/DiffAbund_Males/Maaslin2/", 
    analysis_method = "LM",
    fixed_effects = c("x0_ager_cat", "x0oh01","smoking_exposure_ga"),
    reference = c("smoking_exposure_ga", "Never"),
    transform = "LOG",
    normalization = "TSS",
    correction = "BH",
    standardize = FALSE,
    plot_heatmap = F,
    plot_scatter = F
  )
  

}

genera_names_map_all <- as.data.frame(tax_table(phy_Q1_genus))["Genus"] %>% 
  mutate(feature = make.names(Genus))

maaslin_results_genera <- read_tsv("appendix_results/A4/DiffAbund_Males/Maaslin2/significant_results.tsv") %>% 
  left_join(genera_names_map_all, by = "feature") %>% 
  mutate(feature = Genus) %>% 
  dplyr::select(-Genus)

```

#### Diff abund 4: ALDEx2 (glm)

```{r}
dir.create("appendix_results/A4/DiffAbund_Males/ALDEx2/", recursive = T)

if (!file.exists("appendix_results/A4/DiffAbund_Males/ALDEx2/ALDEx2.DA.rawResults_genera.list.RDS")) {
  
  library(ALDEx2)
  # define model matrix, based on the metadata
  model_mtx <-
    model.matrix(
      ~ x0_ager_cat + x0oh01 + smoking_exposure_ga,
      data = dplyr::select(
        meta(phy_Q1_A4),
        x0oh01,
        x0_ager_cat,
        smoking_exposure_ga
      )
    )
  # perform the clr differential abundance
  aldex.glm.genera_smoking_ga <-
    aldex.clr(abundances(phy_Q1_A4),
              model_mtx,
              mc.samples = 200,
              denom = "all", 
              useMC = T
              )
  
  # make glm, obtain significance of terms
  glm.test_genera_smoking_ga <- aldex.glm(aldex.glm.genera_smoking_ga, model_mtx)
  
  # this step gets estimates effect sizes, it takes ages, like 30-60 minutes
  glm.effects_genera_smoking_ga <- aldex.glm.effect(aldex.glm.genera_smoking_ga, include.sample.summary = T, verbose = FALSE, useMC = T)
  
  # aggregate these results into one data frame, that looks like lme4 output
  aldex.results.genera <- list(clr.model = aldex.glm.genera_smoking_ga,
                        glm.test = glm.test_genera_smoking_ga,
                        eff_size.test = glm.effects_genera_smoking_ga)
  
  
  saveRDS(aldex.results.genera, "appendix_results/A4/DiffAbund_Males/ALDEx2/ALDEx2.DA.rawResults_genera.list.RDS")
  
} else {
  aldex.results.genera <- readRDS("appendix_results/A4/DiffAbund_Males/ALDEx2/ALDEx2.DA.rawResults_genera.list.RDS")
}

```

#### Diff abund 5: ANCOM-BC

```{r}
dir.create("appendix_results/A4/DiffAbund_Males/ANCOM-BC/", recursive = T)

if (!file.exists("appendix_results/A4/DiffAbund_Males/ANCOM-BC/ANCOM-BC.DA.rawResults_genera.list.RDS")) {
tmp <- mia::makeTreeSummarizedExperimentFromPhyloseq(phy_Q1_A4 %>% select_sample_data(x0_sex, x0_ager_cat, x0oh01, smoking_exposure_ga)) 


library(ANCOMBC)

# takes around 30 minutes
ancom_results <- ancombc2(data = tmp,
        assay_name = "counts", 
        tax_level = NULL,
        fix_formula = "x0_ager_cat + x0oh01 + smoking_exposure_ga", 
        group = "smoking_exposure_ga",
        struc_zero = TRUE,
        iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
        em_control = list(tol = 1e-5, max_iter = 20),  
        p_adj_method = "BH",
        n_cl = 1,
        alpha = 0.05)

saveRDS(ancom_results, "appendix_results/A4/DiffAbund_Males/ANCOM-BC/ANCOM-BC.DA.rawResults_genera.list.RDS")
detach("package:ANCOMBC", unload = T)
}else{
  
  ancom_results <- readRDS("appendix_results/A4/DiffAbund_Males/ANCOM-BC/ANCOM-BC.DA.rawResults_genera.list.RDS")
  
}

```

#### Summary of results

```{r}

library(DESeq2)

# DESeq2
deseq2_signif_results <- results(deseq2_results_genus, 
                                 contrast = c("smoking_exposure_ga","Current", "Never"),
                                 pAdjustMethod = "BH") %>% 
  as.data.frame() %>% 
  rownames_to_column("taxon") %>% 
  filter(padj < 0.05)

# LinDA
LinDAresults_taxon_signif_Current <- LinDA_model_genera$output$smoking_exposure_gaCurrent %>% 
  filter(padj < 0.05, 
         #abs(log2FoldChange) > 0.5
         ) %>% 
  rownames_to_column("taxon")

# MaAsLin2
maaslin_signif <- maaslin_results_genera %>% 
  filter(metadata == "smoking_exposure_ga") %>% 
  dplyr::select(feature, value, coef, stderr, pval, qval, N.not.0) %>% 
  filter(qval < 0.05, 
         #abs(coef) > 0.5, 
         value == "Current")

# ALDEx2
aldex_gLinDAsmoking_signif <- aldex.results.genera$glm.test %>% 
  as.data.frame() %>% 
  dplyr::select(contains("smoking_exposure_gaCurrent")) %>% 
  dplyr::rename_all(.funs = function(x) gsub(pattern =  ".*\\:", replacement = "", x = x)) %>% 
  rownames_to_column("taxon") %>% 
  as_tibble() %>% 
  filter(
    #abs(Est) > 0.5, 
    pval.holm < 0.05)
  
# ANCOM-BC

ancom2_signif_results <- ancom_results$res %>% 
  dplyr::select(contains(c("taxon", "gaCurrent")), -contains("diff_")) %>% 
  rownames_to_column("taxon") %>% 
  set_names(c("taxon", "log2FoldChange", "lfcSE", "wald_stat", "p.value", "padj")) %>% 
  filter(
    #abs(log2FoldChange) > 0.5,
    padj < 0.05
    )

all_models_signif_genera <- list(ALDEx2 = aldex_gLinDAsmoking_signif %>% dplyr::select(taxon, Est) %>% set_names(c("taxon", "ALDEx2")), 
                    MaAsLin2 = maaslin_signif %>% dplyr::select(feature, coef) %>% set_names(c("taxon", "MaAsLin2")), 
                    LinDA = LinDAresults_taxon_signif_Current %>% dplyr::select(taxon,log2FoldChange) %>% set_names(c("taxon", "LinDA")), 
                    DESeq2 = deseq2_signif_results %>% dplyr::select(taxon, log2FoldChange) %>% set_names(c("taxon", "DESeq2")),
                    `ANCOM-BC` = ancom2_signif_results %>% dplyr::select(taxon, log2FoldChange) %>% set_names(c("taxon", "ANCOM-BC"))
                    ) %>% 
  purrr::reduce(., full_join, by = "taxon")



frequency_signif_taxon <- pivot_longer(all_models_signif_genera, cols = !starts_with("taxon"), names_to = "DA_method", values_to = "da_eff_size") %>% 
  filter(!is.na(da_eff_size)) %>% 
  group_by(taxon) %>% 
  tally() %>% 
  arrange(desc(n))


taxa_union <- all_models_signif_genera$taxon

taxa_consensus <- frequency_signif_taxon %>% 
  filter(n >= 4) %>% 
  .$taxon

taxa_intersect <- all_models_signif_genera %>% 
  .[complete.cases(.),] %>%
  .$taxon

saveRDS(taxa_union, "appendix_results/A4/DiffAbund_Males/significant_genera_union.RDS")
saveRDS(taxa_intersect, "appendix_results/A4/DiffAbund_Males/significant_genera_intersect.RDS")
saveRDS(taxa_consensus,  "appendix_results/A4/DiffAbund_Males/significant_genera_consensus.RDS")

saveRDS(all_models_signif_genera, "appendix_results/A4/DiffAbund_Males/significant_genera_Consensus-and-log2FC.RDS")
```

### Females {.tabset}

```{r}

phy_Q1_A4 <-filter_sample_data(phy_Q1_genus, x0_sex == "Female")

```

#### Diff abund 1: DESeq2

```{r, fig.align='center'}

dir.create("appendix_results/A4/DiffAbund_Females/DESeq2/", recursive = T)

if(!file.exists("appendix_results/A4/DiffAbund_Females/DESeq2/figureA4_deseq2_dataframe.list.RDS")){
library(DESeq2)
#############################################################################

deseq_object_genus <-
  phyloseq_to_deseq2(phy_Q1_A4, design = ~ x0_ager_cat + x0oh01 + smoking_exposure_ga)

geom_means <- apply(counts(deseq_object_genus), 1, function(x) exp(mean(log(x[x>0]))))

# estimate size factors, important to account for library size variations
estim_size <- estimateSizeFactors(deseq_object_genus, geoMeans = geom_means)

# compute the DESeq2 glm, accounting for the estimate size factor and geometric means

deseq2_results_genus <- DESeq(estim_size, fitType = "local")


saveRDS(deseq2_results_genus, "appendix_results/A4/DiffAbund_Females/DESeq2/figureA4_deseq2_dataframe.list.RDS")
} else{
  deseq2_results_genus <- readRDS("appendix_results/A4/DiffAbund_Females/DESeq2/figureA4_deseq2_dataframe.list.RDS")
}

```

#### Diff abund 2: LinDA

```{r, include=FALSE}
#devtools::install_github("zhouhj1994/LinDA")

dir.create("appendix_results/A4/DiffAbund_Females/LinDA/", recursive = T)

if(!file.exists("appendix_results/A4/DiffAbund_Females/LinDA/LinDA_output.list.RDS")){
  
library(LinDA)

LinDA_model_genera <- linda(otu.tab = abundances(phy_Q1_A4),
                   meta = meta(phy_Q1_A4), 
                   formula =  "~ x0_ager_cat + x0oh01 + smoking_exposure_ga", 
                   alpha = 0.05,
                   prev.cut = 0.01, 
                   lib.cut = 1000, 
                   winsor.quan = 0.97,
                   adaptive = T)

saveRDS(LinDA_model_genera, "appendix_results/A4/DiffAbund_Females/LinDA/LinDA_output.list.RDS")
}else{
  LinDA_model_genera <- readRDS("appendix_results/A4/DiffAbund_Females/LinDA/LinDA_output.list.RDS")
}

```


#### Diff abund 3: Maaslin2

```{r, include=FALSE}

dir.create("appendix_results/A4/DiffAbund_Females/Maaslin2", showWarnings = FALSE, recursive = T)


if(!file.exists("appendix_results/A4/DiffAbund_Females/Maaslin2/significant_results.tsv")){
  
  library(Maaslin2)
 # older normaliz and so on 
  # Maaslin2(
  #   input_data = picrust_taxa_matrix_Q1, 
  #   input_metadata = phy_Q1_meta, 
  #   output = "../results/Maaslin2_DA", 
  #   analysis_method = "LM",
  #   fixed_effects = c("age_cat", "sex", "x0oh01", "smoking_exposure_ga"),
  #   reference = c("smoking_exposure_ga, Never"),
  #   transform = "LOG",
  #   normalization = "TMM",
  #   correction = "BH",
  #   plot_heatmap = F
  # )

  # these following are the same settings used by Nearing  et al. (2021) 
  Maaslin2(
    input_data = abundances(phy_Q1_A4), 
    input_metadata = meta(phy_Q1_A4), 
    output = "appendix_results/A4/DiffAbund_Females/Maaslin2/", 
    analysis_method = "LM",
    fixed_effects = c("x0_ager_cat", "x0oh01","smoking_exposure_ga"),
    reference = c("smoking_exposure_ga", "Never"),
    transform = "LOG",
    normalization = "TSS",
    correction = "BH",
    standardize = FALSE,
    plot_heatmap = F,
    plot_scatter = F
  )
  

}

genera_names_map_all <- as.data.frame(tax_table(phy_Q1_genus))["Genus"] %>% 
  mutate(feature = make.names(Genus))

maaslin_results_genera <- read_tsv("appendix_results/A4/DiffAbund_Females/Maaslin2/significant_results.tsv") %>% 
  left_join(genera_names_map_all, by = "feature") %>% 
  mutate(feature = Genus) %>% 
  dplyr::select(-Genus)

```

#### Diff abund 4: ALDEx2 (glm)

```{r}
dir.create("appendix_results/A4/DiffAbund_Females/ALDEx2/", recursive = T)

if (!file.exists("appendix_results/A4/DiffAbund_Females/ALDEx2/ALDEx2.DA.rawResults_genera.list.RDS")) {
  
  library(ALDEx2)
  # define model matrix, based on the metadata
  model_mtx <-
    model.matrix(
      ~ x0_ager_cat + x0oh01 + smoking_exposure_ga,
      data = dplyr::select(
        meta(phy_Q1_A4),
        x0oh01,
        x0_ager_cat,
        smoking_exposure_ga
      )
    )
  # perform the clr differential abundance
  aldex.glm.genera_smoking_ga <-
    aldex.clr(abundances(phy_Q1_A4),
              model_mtx,
              mc.samples = 200,
              denom = "all", 
              useMC = T
              )
  
  # make glm, obtain significance of terms
  glm.test_genera_smoking_ga <- aldex.glm(aldex.glm.genera_smoking_ga, model_mtx)
  
  # this step gets estimates effect sizes, it takes ages, like 30-60 minutes
  glm.effects_genera_smoking_ga <- aldex.glm.effect(aldex.glm.genera_smoking_ga, include.sample.summary = T, verbose = FALSE, useMC = T)
  
  # aggregate these results into one data frame, that looks like lme4 output
  aldex.results.genera <- list(clr.model = aldex.glm.genera_smoking_ga,
                        glm.test = glm.test_genera_smoking_ga,
                        eff_size.test = glm.effects_genera_smoking_ga)
  
  
  saveRDS(aldex.results.genera, "appendix_results/A4/DiffAbund_Females/ALDEx2/ALDEx2.DA.rawResults_genera.list.RDS")
  
} else {
  aldex.results.genera <- readRDS("appendix_results/A4/DiffAbund_Females/ALDEx2/ALDEx2.DA.rawResults_genera.list.RDS")
}

```

#### Diff abund 5: ANCOM-BC

```{r}
dir.create("appendix_results/A4/DiffAbund_Females/ANCOM-BC/", recursive = T)

if (!file.exists("appendix_results/A4/DiffAbund_Females/ANCOM-BC/ANCOM-BC.DA.rawResults_genera.list.RDS")) {
tmp <- mia::makeTreeSummarizedExperimentFromPhyloseq(phy_Q1_A4 %>% select_sample_data(x0_sex, x0_ager_cat, x0oh01, smoking_exposure_ga)) 


library(ANCOMBC)

# takes around 30 minutes
ancom_results <- ancombc2(data = tmp,
        assay_name = "counts", 
        tax_level = NULL,
        fix_formula = "x0_ager_cat + x0oh01 + smoking_exposure_ga", 
        group = "smoking_exposure_ga",
        struc_zero = TRUE,
        iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
        em_control = list(tol = 1e-5, max_iter = 20),  
        p_adj_method = "BH",
        n_cl = 1,
        alpha = 0.05)

saveRDS(ancom_results, "appendix_results/A4/DiffAbund_Females/ANCOM-BC/ANCOM-BC.DA.rawResults_genera.list.RDS")
detach("package:ANCOMBC", unload = T)
}else{
  
  ancom_results <- readRDS("appendix_results/A4/DiffAbund_Females/ANCOM-BC/ANCOM-BC.DA.rawResults_genera.list.RDS")
  
}

```

#### Summary of results

```{r}

library(DESeq2)

# DESeq2
deseq2_signif_results <- results(deseq2_results_genus, 
                                 contrast = c("smoking_exposure_ga","Current", "Never"),
                                 pAdjustMethod = "BH") %>% 
  as.data.frame() %>% 
  rownames_to_column("taxon") %>% 
  filter(padj < 0.05)

# LinDA
LinDAresults_taxon_signif_Current <- LinDA_model_genera$output$smoking_exposure_gaCurrent %>% 
  filter(padj < 0.05, 
         #abs(log2FoldChange) > 0.5
         ) %>% 
  rownames_to_column("taxon")

# MaAsLin2
maaslin_signif <- maaslin_results_genera %>% 
  filter(metadata == "smoking_exposure_ga") %>% 
  dplyr::select(feature, value, coef, stderr, pval, qval, N.not.0) %>% 
  filter(qval < 0.05, 
         #abs(coef) > 0.5, 
         value == "Current")

# ALDEx2
aldex_gLinDAsmoking_signif <- aldex.results.genera$glm.test %>% 
  as.data.frame() %>% 
  dplyr::select(contains("smoking_exposure_gaCurrent")) %>% 
  dplyr::rename_all(.funs = function(x) gsub(pattern =  ".*\\:", replacement = "", x = x)) %>% 
  rownames_to_column("taxon") %>% 
  as_tibble() %>% 
  filter(
    #abs(Est) > 0.5, 
    pval.holm < 0.05)
  
# ANCOM-BC

ancom2_signif_results <- ancom_results$res %>% 
  dplyr::select(contains(c("taxon", "gaCurrent")), -contains("diff_")) %>% 
  rownames_to_column("taxon") %>% 
  set_names(c("taxon", "log2FoldChange", "lfcSE", "wald_stat", "p.value", "padj")) %>% 
  filter(
    #abs(log2FoldChange) > 0.5,
    padj < 0.05
    )

all_models_signif_genera <- list(ALDEx2 = aldex_gLinDAsmoking_signif %>% dplyr::select(taxon, Est) %>% set_names(c("taxon", "ALDEx2")), 
                    MaAsLin2 = maaslin_signif %>% dplyr::select(feature, coef) %>% set_names(c("taxon", "MaAsLin2")), 
                    LinDA = LinDAresults_taxon_signif_Current %>% dplyr::select(taxon,log2FoldChange) %>% set_names(c("taxon", "LinDA")), 
                    DESeq2 = deseq2_signif_results %>% dplyr::select(taxon, log2FoldChange) %>% set_names(c("taxon", "DESeq2")),
                    `ANCOM-BC` = ancom2_signif_results %>% dplyr::select(taxon, log2FoldChange) %>% set_names(c("taxon", "ANCOM-BC"))
                    ) %>% 
  purrr::reduce(., full_join, by = "taxon")



frequency_signif_taxon <- pivot_longer(all_models_signif_genera, cols = !starts_with("taxon"), names_to = "DA_method", values_to = "da_eff_size") %>% 
  filter(!is.na(da_eff_size)) %>% 
  group_by(taxon) %>% 
  tally() %>% 
  arrange(desc(n))


taxa_union <- all_models_signif_genera$taxon

taxa_consensus <- frequency_signif_taxon %>% 
  filter(n >= 4) %>% 
  .$taxon

taxa_intersect <- all_models_signif_genera %>% 
  .[complete.cases(.),] %>%
  .$taxon

saveRDS(taxa_union, "appendix_results/A4/DiffAbund_Females/significant_genera_union.RDS")
saveRDS(taxa_intersect, "appendix_results/A4/DiffAbund_Females/significant_genera_intersect.RDS")
saveRDS(taxa_consensus,  "appendix_results/A4/DiffAbund_Females/significant_genera_consensus.RDS")

saveRDS(all_models_signif_genera, "appendix_results/A4/DiffAbund_Females/significant_genera_Consensus-and-log2FC.RDS")
```

## Differences between females and males?

Benchmarking with the analysis in the main body

```{r}

males_signif_genera <- readRDS("appendix_results/A4/DiffAbund_Males/significant_genera_Consensus-and-log2FC.RDS")

females_signif_genera <- readRDS("appendix_results/A4/DiffAbund_Females/significant_genera_Consensus-and-log2FC.RDS")

figure1_genera <- readRDS("../Main/results/Figure1/DiffAbundGenera/significant_genera_Consensus-and-log2FC.RDS")

# make a gradient of concordance of significant genera between females and males

males_genera_count_signif <- males_signif_genera %>% 
  pivot_longer(cols = colnames(males_signif_genera)[-1], 
               names_to = "method", 
               values_to = "values") %>% 
  filter(!is.na(values)) %>% 
  group_by(taxon) %>% 
  summarise(n_signif = n(),
            mean_lfc = mean(values, na.rm = T)) %>% 
  mutate(group = "Males Only")

females_genera_count_signif <- females_signif_genera %>% 
  pivot_longer(cols = colnames(females_signif_genera)[-1], 
               names_to = "method", 
               values_to = "values") %>% 
  filter(!is.na(values)) %>% 
  group_by(taxon) %>% 
  summarise(n_signif = n(),
            mean_lfc = mean(values, na.rm = T)) %>% 
  mutate(group = "Females Only")

model_paper_count_signif <- figure1_genera %>% 
  pivot_longer(cols = colnames(figure1_genera)[-1], 
               names_to = "method", 
               values_to = "values") %>% 
  filter(!is.na(values)) %>% 
  group_by(taxon) %>% 
  summarise(n_signif = n(),
            mean_lfc = mean(values, na.rm = T)) %>% 
  mutate(group = "Sex Adjusted")
```

```{r}
data_unified_for_plot_males <- list(
  males_genera_count_signif,
  model_paper_count_signif
  )

intersection_signif_taxa <- sapply(1:5, function(n) data_unified_for_plot_males %>% 
         lapply(filter, n_signif >= n) %>% 
         lapply("[[", "taxon") %>% 
         purrr::reduce(intersect) %>% 
         length()
       
       )

union_signif_taxa <- sapply(1:5, function(n) data_unified_for_plot_males %>% 
         lapply(filter, n_signif >= n) %>% 
         lapply("[[", "taxon") %>% 
         purrr::reduce(union) %>% 
         length()
       
       )

intersection_vs_union_males <- data.frame(x = factor(paste0(1:5, "/5")), 
           y = intersection_signif_taxa/union_signif_taxa*100) 

###### same but for females

data_unified_for_plot_females <- list(
  females_genera_count_signif,
  model_paper_count_signif
  )

intersection_signif_taxa <- sapply(1:5, function(n) data_unified_for_plot_females %>% 
         lapply(filter, n_signif >= n) %>% 
         lapply("[[", "taxon") %>% 
         purrr::reduce(intersect) %>% 
         length()
       
       )

union_signif_taxa <- sapply(1:5, function(n) data_unified_for_plot_females %>% 
         lapply(filter, n_signif >= n) %>% 
         lapply("[[", "taxon") %>% 
         purrr::reduce(union) %>% 
         length()
       
       )

intersection_vs_union_females <- data.frame(x = factor(paste0(1:5, "/5")), 
           y = intersection_signif_taxa/union_signif_taxa*100)
```

#### Venn diagram at 4/5 DA methods

**Concordance of the significant genera in each of the three models.** Genera significant over 4/5 methods were considered significant overall. the three models were designed as follows:

  - `Males Only` = $Genera \sim Age~Group + N° teeth + Smoking~status$ only on male participants
  - `Females Only` = $Genera \sim Age~Group + N° teeth + Smoking~status$ only on female participants
  - `Sex Adjusted` = $Genera \sim Age~Group + N° teeth + \color{red}{\textbf{Sex}} + Smoking~status$ on all participants
  
Numbers in the venn diagram slots are the number of genera in common between the models.

```{r}
data_unified_for_plot_all <- list(
  "Males Only" = males_genera_count_signif,
  "Sex Adjusted" = model_paper_count_signif,
  "Females Only" = females_genera_count_signif
  
  )

venn_Diagr_signif_4of5 <- data_unified_for_plot_all %>% 
  lapply(filter, n_signif >= 4) %>% 
  lapply("[[", "taxon") %>% 
  ggVennDiagram::ggVennDiagram(edge_size = 0)+ 
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_distiller(palette = "RdBu")+
  scale_color_manual(values = rep("darkgray", 3))

venn_Diagr_signif_4of5
```

**Figure A4. Some genera are exclusive of males and females, but are all found in the sex-adjusted significant genera*.* **(A)** Correlation between the 23 genera found always significant (see panel **C**). **(B)** Proportion (%) of concordance of the genera found significant in only Males and only Females in relation to smoking in comparison with those found in the fixed effect model. The x axis represent the minimum number of methods that a genus should be significant to be considered consensus-wide significant. **(C)** Venn diagram of the three models used. in all cases, the contrast chosen was Current/Never, adjusting for age and number of teeth. All significance thresholds were q-values < 0.05, all Benjamini-Hochberg (5% FDR) except for ALDEx2, which uses Holm correction.

```{r}
plot_A5 <- ggarrange(plot_correlations,
                     ggarrange(plot_concordance_vs_threshold +theme(legend.position = "top"),
          venn_Diagr_signif_4of5 + theme(legend.position = "none"), 
          ncol = 1, 
          nrow = 2,
          labels =  c("B", "C")
          ), 
          nrow = 1, ncol = 2, labels = c("A", ""))

ggsave(plot = plot_A5, 
       filename = "appendix_results/A5/A5_concordance_sex_separated_vs_sex_adjusted.png", 
       height = 6, 
       width = 12, 
       dpi = 300,
       bg = "white")

ggsave(plot = plot_importance_method, 
       filename = "appendix_results/A5/A5b_importance_of_each_method_in_finding_a_signal.png", 
       height = 5, 
       width = 8, 
       dpi = 300,
       bg = "white")


plot_A5
```

**Figure A5 extra. Benchmark of the consensus-based differential abundance analysis considering the proportion of genera found significant in 3 out of 4 methods by bootstrapping one method at a time and comparing the set of significant genera with the set found in 4 out of 5 methods. The method name that colors the lines is the one removed in that round of bootstrap.

```{r}
plot_importance_method
```

## Figure A6 - Correlations log2FC diff. abund vs log2FC regression on daily smoking intensity

**Figure A6. Scatterplot and correlation index between effect size estimates in the differential abundance contrasting Current over Never smokers and the effect size estimates for the regression of genera abundance in relation to the daily grams of tobacco smoked daily**. Both methods used DESeq2 regression correcting for age group, sex, number of teeth. , however in the comparison between Current and Never smokers a contrast between factors was extract (log2 Fold Change), in the second case the estimates were step-wise fold change direction in response to increasing 5 grams smoked per day (0, 5, 10, ..., 30).

```{r}

q1_results <- readRDS("/home/gantonello/CHRISMB/PAPER_DRAFTS/smoking-paper-genus/Main/results/Figure1/DiffAbundGenera/significant_genera_Consensus-and-log2FC.RDS") %>% 
  dplyr::filter(taxon %in% readRDS("/home/gantonello/CHRISMB/PAPER_DRAFTS/smoking-paper-genus/Main/results/Figure1/DiffAbundGenera/significant_genera_consensus.RDS")) %>% 
  column_to_rownames("taxon") %>% 
  rowMeans(na.rm = T) %>% 
  as.data.frame() %>% 
  rownames_to_column("Genus") %>% 
  set_names(c("Genus", "Q1_log2FC"))

q2_results <- readRDS("/home/gantonello/CHRISMB/PAPER_DRAFTS/smoking-paper-genus/Main/results/Figure2/deseq_smoking_regression_datafames.rds") %>% 
  .$x0sm61_bin5cont %>% 
  as.data.frame() %>% 
  filter(padj < 0.05)

data_for_plot <- inner_join(q1_results, q2_results, by = "Genus")


############# actual plot and significance 

pearson_rho <- cor.test(data_for_plot$Q1_log2FC, data_for_plot$log2FoldChange, method = "pearson")
library(RColorBrewer)

fig.A6 <- data_for_plot %>%
  
  ggplot(aes(x = log2FoldChange, y = Q1_log2FC, color = Phylum, shape = Phylum))+
  geom_point(size = 2)+
  geom_hline(yintercept = 0, lty = "dashed", color = "darkgray")+
  geom_vline(xintercept = 0, lty = "dashed", color = "darkgray")+
  labs(
       y = "Eff. Sizes Smokers vs Non-Smokers (Differential Abundance)", 
       x = "Eff. Sizes Daily Smoking Intensity (Regression)",
       subtitle = paste("Pearson's rho:", round(pearson_rho$estimate, 3), "| P-value:", format(pearson_rho$p.value, scientific = TRUE, digits = 2)))+
  scale_color_brewer(palette = "Dark2")


ggsave(plot = fig.A6, 
       filename = "appendix_results/A6/A6_eff_sizes_DESeq2_proportional.png",
       width = 7, 
       height = 5, 
       dpi = 300
       )

fig.A6

```

## Figure A7 - Gram staining relative abundances vs Smoking  habits

For the sake of this observation, I manually curated a table of the phyla I had in my taxonomic table, since Gram staining is generally discriminative at phylum level. Then I plotted the proportions of Gram negative and positives in relation to smoking. Given the similar profiles as aerobes, I think the key here is that since Proteobacteria (G -) decrease and Actinobacteria and several Firmicutes (G+) increase, this is why we see similar profiles as aerobes/anaerobes as in Figure A3.

```{r, height = 5, width = 6}

gram <- read_tsv("appendix-input-data/gram_staining_table.tsv")

taxtab <- as.data.frame(tax_table(phy_Q1)) %>% 
  dplyr::select(ASV, Phylum)

otu_tab <- phy_Q1 %>% 
  phy_transform("compositional") %>% 
  abundances() %>% 
  as.data.frame() %>% 
  rownames_to_column("ASV")

tmp <- left_join(taxtab, gram, by = "Phylum") %>% 
  left_join(otu_tab, by = "ASV") %>% 
  dplyr::select(-ASV, -Phylum) %>% 
  group_split(Gram) %>% 
  set_names(c("Gram_neg", "Gram_pos")) %>% 
  lapply(function(df) dplyr::select(df, -Gram)) %>% 
  sapply(colSums) %>%
  as.data.frame() %>%
  rownames_to_column("sample_name")

phy_Q1_gram <- phy_add_metadata_variables(physeq = phy_Q1, df = tmp, by = "sample_name") 

signif <- 
  phy_Q1_gram %>% 
  meta() %>% 
  dplyr::select(sample_name,x0sm51, Gram_pos, Gram_neg) %>% 
  pivot_longer(cols = c("Gram_pos", "Gram_neg"), names_to =  "gram_staining") %>% 
  
  group_by(gram_staining) %>% 
  rstatix::wilcox_test(value ~ x0sm51) %>% 
  rstatix::adjust_pvalue(method = "BH") %>% 
  rstatix::add_xy_position(x = "gram_staining",step.increase = 0.2, fun = "max") %>% 
  filter(p.adj < 0.05)

basic_plot <- phy_Q1_gram %>% 
  meta() %>% 
  dplyr::select(sample_name, x0sm51, Gram_pos, Gram_neg) %>% 
  pivot_longer(cols = c("Gram_pos", "Gram_neg"), names_to =  "gram_staining") %>% 
  mutate(gram_staining = ifelse(gram_staining == "Gram_pos", "Gram +", "Gram -")) %>% 
  ggpubr::ggboxplot(x = "gram_staining", 
                    y = "value", 
                    fill = "x0sm51")+
  theme_light()

fig_A6 <- basic_plot + 
  stat_pvalue_manual(signif, tip.length = 0.01)+
  labs(
    x = "Gram Staining",
    y = "Relative abundance",
    fill = "Smoking Habit")+
  scale_fill_manual(values = ggsci::pal_jco()(4)[c(1,2,4,3)])


ggsave(plot = fig_A6,
       filename = "appendix_results/A6/A6_gram_staining_vs_smokers.png",
       height = 5, 
       width = 7,
       dpi = 300
       )

fig_A6

```



## Figure A8 - Correlation of differentially abundant pathways

```{r}
picrust_pathways_matrix <- data.table::fread("/home/gantonello/CHRISMB/PICRUSt2_prediction_v2.5/unstratified/picrust2_output/pathways_out/path_abun_unstrat_descrip.tsv.gz") %>% 
  dplyr::select(-pathway) %>% 
  column_to_rownames("description") %>% 
  as.matrix()

picrust_pathways_matrix %<>% round(0)

signif_pathways <- readRDS("../Main/results/Figure4/significant_PWYs_consensus.RDS")

picrust_pathways_matrix_Q1 <- picrust_pathways_matrix[signif_pathways, sample_names(phy_Q1)] 
picrust_pathways_matrix_Q1%<>% apply(1, function(x) x/sum(x)) %>% 
  t()
```

```{r, include=FALSE}

correl_mats <- Hmisc::rcorr(t(picrust_pathways_matrix_Q1), type = "spearman")
png(filename = "appendix_results/A8/A8_correlation_significant_pathways.png",
    height = 12, width = 14, units = "in", res = 300)

corrplot::corrplot(correl_mats$r, method = "number", type = "lower", tl.col = 'black', tl.srt = 45, p.mat = corrplot::cor.mtest(t(picrust_pathways_matrix_Q1))$p, sig.level = 0.01)

dev.off()
```

**Figure A8. Correlation among pathways that were found differentially abundant in smokers compared to never smokers.** Pathways were transformed to relative abundance to account for compositionality, and a pairwise Spearman correlation matrix was calculated with the `Hmisc::rcorr` function, which additionally reports a P-value as indication of the strength of the correlation estimate. Blue circles are **positive** correlations, red circles are **negative** correlations. Larger circles mean a stronger correlation. Crossed-out slots are non-significant associations (p >= 0.01).

```{r, fig.height=14, fig.width=14}
corrplot::corrplot(correl_mats$r, method = "number", type = "lower", tl.col = 'black', tl.srt = 45, p.mat = corrplot::cor.mtest(t(picrust_pathways_matrix_Q1))$p, sig.level = 0.01)

```

# Supplementary Tables

## Table A1 - Literature review, table of ethnicities and sample size of studies

```{r, results="asis"}
tab.A1 <- read_tsv("appendix-input-data/Table A1 - literature ethnicities and sample sizes.tsv") %>% 
  mutate(Reference = gsub(pattern = "al.,", replacement = "al.", x = Reference, fixed = T) %>% gsub("\\(|\\)", "", .))

tab.A1 %>% 
  dplyr::rename(`Reference link`= DOI) %>% 
  #mutate(Reference = cell_spec(Reference, format = "html", link = DOI)) %>% 
  #select(-DOI) %>% 
kbl(caption = "Comparison of sample size, ethnicity and country of residence of participants in previous studies on the relationship between smoking and the salivary microbiota.") %>% 
      kable_styling()

```

## Table A2 - eHOMD database expansion, list of genomes accession numbers per species

```{r}

tableS2 <- read_tsv("appendix-input-data/extra_eHOMD_reference_sequences.tsv") %>% 
  set_names(c("Species", "NCBI Reference Sequences")) %>% 
  mutate(Species = paste0("*", Species, "*"))

kbl(tableS2, caption = "Genome accession numbers (NCBI codes) for each species. The 16S rRNA genes of each genome were added to the locally downloaded extended Human Oral Microbiome Database (eHOMD) in order to maximize the likelihood of species level taxonomic assignment") %>% 
  kable_styling(full_width = T) 
  
```

## Table A3 - Sample exclusion steps

```{r}

col1 <- c(
  "None",
  "Missing Smoking Status",
  "Missing N° of Teeth",
  "Missing Antibiotic Usage Information",
  "Took Antibiotics in the last 3 months"
)

col3 <- c( 
  nsamples(chrismb_phy_core),
  
  chrismb_phy_core %>% 
  filter_sample_data(!(AID %in% excluded_samples_upon_request),
                     !is.na(x0sm51)) %>% 
    nsamples(),
  
  chrismb_phy_core %>% 
  filter_sample_data(!(AID %in% excluded_samples_upon_request),
                     !is.na(x0sm51),
                     !is.na(x0oh01)) %>% 
    nsamples(),
  
  chrismb_phy_core %>% 
  filter_sample_data(!(AID %in% excluded_samples_upon_request),
                     !is.na(x0sm51),
                     !is.na(x0oh01),
                     !is.na(x0dd51)
                     ) %>% 
    nsamples(),
  
  chrismb_phy_core %>% 
  filter_sample_data(!(AID %in% excluded_samples_upon_request),
                     !is.na(x0sm51),
                     !is.na(x0oh01),
                     !is.na(x0dd51),
                     x0dd51 != "Yes"
                     ) %>% 
    nsamples()
) 

col2 <- c(0, abs(diff(col3)))

matrix_table_samples_excluded <- cbind(
  col1, 
  col2, 
  col3
) %>% 
  set_colnames(c("Condition", "Samples Excluded (N)", "Samples after exclusion (N)"))


kbl(matrix_table_samples_excluded) %>% 
  kable_styling()
```


## Table A4 - Population differences between excluded and included participants for the study

```{r, fig.cap="Differences between excluded and included participants in the study. Significance (p-value column) was calculated as Welch's t-test for continuous variables, Fisher exact test for 2x2 comparisons, and Pearson's Chi squared test for multiclass factors. Abbreviations used: BP = Blood pressure; BMI = Body Mass Index"}

df <- chrismb_phy %>% 
  meta() %>% 
  mutate(included = 
           case_when(AID %in% meta(phy_Q1)[["AID"]] ~ 'included', TRUE ~ 'excluded')
         )
#, then use gtsummary to do a summary table by your variables of interest, i.e. 

library(gtsummary)


tbl <- df %>% 
  select(any_of(c("x0sm51", "x0dd51", "x0bp01", "x0bp02", "x0an03a", "x0oh01", "x0oh02b", "included"))) %>% 
  set_names(c("Smoking Habit", "Antibiotics Usage", "Systolic BP", "Diastolic BP", "BMI", "N° Teeth", "Gum Health (self-assessed)", "included")) %>% 
  tbl_summary(by=included, statistic = list(all_continuous()~"{mean} ({sd})")) %>%
  add_p(test=list(all_continuous() ~ "t.test"))  %>%
  as_kable_extra() %>% 
  kable_styling()

tbl
```

## Table A5 - Per taxonomic level microbiota statistics

**Table A5. CHRISMB total reads and total taxonomic ranks before and after filtering by prevalence and detection with the function 'microbiome::core'.**

```{r, echo=FALSE}

tax_lvls <- colnames(chrismb_phy@tax_table) 
  
tmp <- rbind(c("-|-", paste(threshold_prevalence,threshold_detection, sep = "|")),
             c(format(sum(sample_sums(chrismb_phy)), scientific = T,digits = 3),format(sum(sample_sums(chrismb_phy_core)), scientific = T,digits = 3)),
             cbind(sapply(tax_lvls,
                          function(lvl)
                            tryCatch(ntaxa(tax_glom(chrismb_phy, lvl)), error = function(e) return(0))),
             sapply(tax_lvls,
             function(lvl)
               tryCatch(ntaxa(tax_glom(chrismb_phy_core, lvl)), error = function(e) return(0)))
             ))

rownames(tmp) <- c(
  "Thresholds of Prevalence|Detection",
  "Number of Reads",
  tax_lvls)

kbl(tmp) %>% 
  kable_styling()

```

## Table A6 - PERMANOVA table of variables included in the model (marginal effect)

```{r, include = FALSE}
distance_bray <- phyloseq::distance(phy_Q1_genus %>% phy_transform("compositional"), "bray")

library(vegan)

betadisper_smoking_bray <- betadisper(distance_bray, group = phy_Q1@sam_data$smoking_exposure_ga)

betadisper_smoking_bray_permutest <-  permutest(betadisper_smoking_bray, 2000) # tukey post-hoc test

permanova_smoking_bray <- adonis2(distance_bray ~ x0oh01 + x0_sex + x0_ager_cat + smoking_exposure_ga, data = meta(phy_Q1), by = "margin", permutations = 2000)
```

**Table A6. Permutational (x2000) Multivariate Analysis of variance (PERMANOVA) of the marginal effect of age, sex, smoking and number of teeth in the compositional variability of the microbiota.** Interindividual microbiota-based dissimilarity was calculated with the Bray-Curtis method on counts transformed to relative abundances (Total Sum Scaling, ranging from 0 to 1). The Beta dispersity (Tuckey post-hoc test significance) was P = `r round(betadisper_smoking_bray_permutest$tab[,ncol(betadisper_smoking_bray_permutest$tab)][1], 3)`.

```{r}
as.data.frame(permanova_smoking_bray) %>% 
  set_rownames(c("Number of Teeth", "Sex", "Age Category", "Smoking habits (3 cat)", "Residuals", "Total")) %>% 
  mutate(R2 = R2*100) %>% 
  set_colnames(c("Deg.Freedom", "SumOfSquares", "R2 (%)", "F-stat", "P-value")) %>%
  rownames_to_column("Term") %>% 
  mutate_if(is.numeric, .funs = function(x)
      ifelse(x < 0.001, 
                 yes = format(x, digits= 2, scientific = T),
                 no = round(x, 1))
      ) %>% 
  kbl() %>% 
  kable_styling()
```

## Table A7 - Linear model aerobes against smoking intensity per day

**Table A7. The relative abundance of AEROBIC bacteria decreases with increasing grams of tobacco smoked in a day.** The trend is true up to 10 g/day, after which a plateau is reached.

```{r}

oxygen_molten_metadata <- readRDS("../Main/results/Figure1/OxygenMetabolism/oxygen_with_metadata_df.RDS")

filter(
    oxygen_molten_metadata,
    oxygen == "Aerobic" &
      smoking_exposure_ga == "Current",
    x0sm61 < 60
  ) %>% 
  lm(value ~ I(1/x0sm61) + x0_ager + x0_sex + x0oh01, data = .) %>% 
  summary() %>% 
  broom::tidy() %>%
  mutate(term = c("Intercept", "1/tobacco exposure intensity (g/day)", "Age", "Sex", "1-9 teeth", "10-19 teeth", "20+ teeth")) %>% 
  kbl(digits = 3) %>% 
  kable_styling()

```

## Table A8 - Linear model anaerobes against smoking intensity per day

**Table A8. The relative abundance of ANAEROBIC bacteria decreases with increasing grams of tobacco smoked per day.** The trend is true up to 10 g/day, after which a plateau is reached.

```{r}

oxygen_molten_metadata <- readRDS("../Main/results/Figure1/OxygenMetabolism/oxygen_with_metadata_df.RDS")

filter(
    oxygen_molten_metadata,
    oxygen == "Anaerobic" &
      smoking_exposure_ga == "Current",
    x0sm61 < 60
  ) %>% 
  lm(value ~ I(1/x0sm61) + x0_ager + x0_sex + x0oh01, data = .) %>% 
  summary() %>% 
  broom::tidy() %>%
  mutate(term = c("Intercept", "1/tobacco exposure intensity (g/day)", "Age", "Sex", "1-9 teeth", "10-19 teeth", "20+ teeth")) %>% 
  kbl(digits = 3) %>% 
  kable_styling()

```

## Table A9 - Linear model aerobes against years since smoking cessation

**The relative abundance of AEROBIC bacteria increases with increasing years since smoking cessation.** The trend reaches a plateau after 20 years, it is no longer statistically associated (data not shown)

```{r}
filter(
    oxygen_molten_metadata,
    oxygen == "Aerobic" &
        sample_name %in% sample_names(phy_Q3_genus) &
        years_since_quitting_or_reducing <= 20) %>% 
    lm(value ~ years_since_quitting_or_reducing + x0_ager + x0_sex + x0oh01, data = .) %>% 
  summary() %>% 
  broom::tidy() %>%
  mutate(term = c("Intercept", "Years since quitting", "Age", "Sex", "1-9 teeth", "10-19 teeth", "20+ teeth")) %>% 
  kbl(digits = 3) %>% 
  kable_styling()
```

## Table A10 - Linear model anaerobes against years since smoking cessation

**The relative abundance of ANAEROBIC bacteria increases with increasing years since smoking cessation.** The trend reaches a plateau after 20 years, it is no longer statistically associated (data not shown)

```{r}
filter(
    oxygen_molten_metadata,
    oxygen == "Anaerobic" &
        sample_name %in% sample_names(phy_Q3_genus) &
        years_since_quitting_or_reducing <= 20) %>% 
    lm(value ~ years_since_quitting_or_reducing + x0_ager + x0_sex + x0oh01, data = .) %>% 
  summary() %>% 
  broom::tidy() %>%
  mutate(term = c("Intercept", "Years since quitting", "Age", "Sex", "1-9 teeth", "10-19 teeth", "20+ teeth")) %>% 
  kbl(digits = 3) %>% 
  kable_styling()
```
